# Introduction to NGS Data Analysis Workshop

---
**The University of Sydney, NSW - 27th - 29th June 2017**


## Instructors

* Katherine Champ (KC) - Bioplatforms Australia, Sydney
* Sonika Tyagi (ST) - AGRF, Melbourne
* Matthew Field (MF) - Australian National University/James Cook University, Cairns
* Xi (Sean) Li (SL) - Australian National University, Canberra
* Susan Corley (SC) - UNSW Systems Biology Initiative, Sydney


## Timetable
### Day 1 - Introduction to the command line, data quality & alignment & ChIP-Seq
**27th June** - [Computer Lab 1.4, Charles Perkins Centre, University of Sydney, NSW](https://goo.gl/maps/feoXLjEhke52)

| **Time** | **Topic** | **Links** | **Instructor** |
| -------- | --------- | --------- | ----------- |
|09:00|Introductions and course orientation||K|
|09:45|Practical: Introduction to the command line|||
|10:15|**Morning Tea**|
|10:40|Practical: Introduction to the command line course and R course|||
|11:20|Introduction to NGS- technology, data formats and introduction to quality control|||
|12:30|**Lunch**|
|13:15|Quality control: Intro to practical|||
|13:25|Practical: Quality control|||
|14:05|Introduction to sequence alignment|||
|14:15|Practical: Sequence alignment|||
|15:00|**Afternoon Tea**|
|15:25|Introduction to ChIP-Seq|||
|15:55|Practical: ChIP-Seq analysis - Peak calling and annotation|||
|16:30|Q&A and day 1 wrap-up||All|


### Day 2 - ChIP-Seq motif analysis and RNA-Seq analysis
**28th June** - [Computer Lab 1.4, Charles Perkins Centre, University of Sydney, NSW](https://goo.gl/maps/feoXLjEhke52)

| **Time** | **Topic** | **Links** | **Instructor** |
| -------- | --------- | --------- | ----------- |
|09:00|Practical: Motif analysis|||
|09:40|Introduction to RNA-Seq|||
|10:30|**Morning Tea**|
|10:50|Practical: Alignment and splice junction identification|||
|12:30|**Lunch**|
|13:30|Practical: Differential gene expression with Bio-conductor package: EdgeR and Voom|||
|15:00|**Afternoon Tea**|
|15:30|Practical: Biological interpretation|||
|16:30|Q&A and day 2 wrap-up||All|


### Day 3 - *de novo* Assembly
**29th June** - [Computer Lab 1.4, Charles Perkins Centre, University of Sydney, NSW](https://goo.gl/maps/feoXLjEhke52)

| **Time** | **Topic** | **Links** | **Instructor** |
| -------- | --------- | --------- | ----------- |
|09:00|Introduction to de novo assembly|||
|09:40|Practical: de novo assembly using Illumina reads|||
|10:30|**Morning Tea**|
|10:50|Practical: de novo assembly using Illumina reads (cont.)|||
|11:30|Practical: de novo assembly using PacBio – Canu workflow|||
|12:30|**Lunch**|
|13:30|Practical: de novo assembly using PacBio – Canu workflow|||
|15:30|**Afternoon Tea**|
|15:50|Practical: Polishing PacBio de novo assembly with Illumina reads|||
|16:30|Q&A and workshop wrap-up||All|
|17:00|Workshop Survey|||
