
<!DOCTYPE html>
<html lang="en" class="no-js">
  <head>
    
      <meta charset="utf-8">
      <meta name="viewport" content="width=device-width,initial-scale=1">
      <meta http-equiv="x-ua-compatible" content="ie=edge">
      
        <meta name="description" content="This contains all material for Markdown version of BTP workshops.">
      
      
      
        <meta name="author" content="Authors">
      
      
        <link rel="shortcut icon" href="../../../assets/images/favicon.png">
      
      <meta name="generator" content="mkdocs-0.16.3, mkdocs-material-1.7.4">
    
    
      
        <title>DeNovo Canu - Markdown Manuals for Bioinformatics Training Platform (BTP)</title>
      
    
    
      <script src="../../../assets/javascripts/modernizr-1df76c4e58.js"></script>
    
    
      <link rel="stylesheet" href="../../../assets/stylesheets/application-769c285a91.css">
      
        <link rel="stylesheet" href="../../../assets/stylesheets/application-02c2a4388f.palette.css">
      
    
    
      
        
        
        
        <link rel="stylesheet" href="https://fonts.googleapis.com/css?family=Roboto:300,400,400i,700|Roboto+Mono">
        <style>body,input{font-family:"Roboto","Helvetica Neue",Helvetica,Arial,sans-serif}code,kbd,pre{font-family:"Roboto Mono","Courier New",Courier,monospace}</style>
      
      <link rel="stylesheet" href="https://fonts.googleapis.com/icon?family=Material+Icons">
    
    
      <link rel="stylesheet" href="../../../css/details.css">
    
    
  </head>
  
  
  
  
    <body data-md-color-primary="indigo" data-md-color-accent="light-blue">
  
    <svg class="md-svg">
      <defs>
        
        
      </defs>
    </svg>
    <input class="md-toggle" data-md-toggle="drawer" type="checkbox" id="drawer">
    <input class="md-toggle" data-md-toggle="search" type="checkbox" id="search">
    <label class="md-overlay" data-md-component="overlay" for="drawer"></label>
    
      <header class="md-header" data-md-component="header">
  <nav class="md-header-nav md-grid">
    <div class="md-flex">
      <div class="md-flex__cell md-flex__cell--shrink">
        
          <a href="../../.." title="Markdown Manuals for Bioinformatics Training Platform (BTP)" class="md-icon md-icon--home md-header-nav__button">
          </a>
        
      </div>
      <div class="md-flex__cell md-flex__cell--shrink">
        <label class="md-icon md-icon--menu md-header-nav__button" for="drawer"></label>
      </div>
      <div class="md-flex__cell md-flex__cell--stretch">
        <span class="md-flex__ellipsis md-header-nav__title">
          
            
              
                <span class="md-header-nav__parent">
                  Introduction to NGS Modules
                </span>
              
            
            DeNovo Canu
          
        </span>
      </div>
      <div class="md-flex__cell md-flex__cell--shrink">
        
          <label class="md-icon md-icon--search md-header-nav__button" for="search"></label>
          
<div class="md-search" data-md-component="search">
  <label class="md-search__overlay" for="search"></label>
  <div class="md-search__inner">
    <form class="md-search__form" name="search">
      <input type="text" class="md-search__input" name="query" required placeholder="Search" accesskey="s" autocapitalize="off" autocorrect="off" autocomplete="off" spellcheck="false" data-md-component="query">
      <label class="md-icon md-search__icon" for="search"></label>
      <button type="reset" class="md-icon md-search__icon" data-md-component="reset">close</button>
    </form>
    <div class="md-search__output">
      <div class="md-search__scrollwrap" data-md-scrollfix>
        <div class="md-search-result" data-md-component="result" data-md-lang-search="">
          <div class="md-search-result__meta" data-md-lang-result-none="No matching documents" data-md-lang-result-one="1 matching document" data-md-lang-result-other="# matching documents">
            Type to start searching
          </div>
          <ol class="md-search-result__list"></ol>
        </div>
      </div>
    </div>
  </div>
</div>
        
      </div>
      
    </div>
  </nav>
</header>
    
    <div class="md-container">
      
      
        

  

<nav class="md-tabs md-tabs--active" data-md-component="tabs">
  <div class="md-tabs__inner md-grid">
    <ul class="md-tabs__list">
      
        
  <li class="md-tabs__item">
    
      <a href="../../.." title="Home" class="md-tabs__link">
        Home
      </a>
    
  </li>

      
        
      
        
  
  
    <li class="md-tabs__item">
      
        <a href="../../../timetables/timetable_introNGS/" title="Timetable" class="md-tabs__link">
          Timetable
        </a>
      
    </li>
  

      
        
      
        
  
  
    <li class="md-tabs__item">
      
        <a href="../../btp-module-ngs-cli/commandline/" title="Introduction to NGS Modules" class="md-tabs__link md-tabs__link--active">
          Introduction to NGS Modules
        </a>
      
    </li>
  

      
    </ul>
  </div>
</nav>
      
      <main class="md-main">
        <div class="md-main__inner md-grid" data-md-component="container">
          
            
              <div class="md-sidebar md-sidebar--primary" data-md-component="navigation">
                <div class="md-sidebar__scrollwrap">
                  <div class="md-sidebar__inner">
                    <nav class="md-nav md-nav--primary" data-md-level="0">
  <label class="md-nav__title md-nav__title--site" for="drawer">
    
      <i class="md-icon md-icon--home md-nav__button"></i>
    
    Markdown Manuals for Bioinformatics Training Platform (BTP)
  </label>
  
  <ul class="md-nav__list" data-md-scrollfix>
    
      
      
      


  <li class="md-nav__item">
    <a href="../../.." title="Home" class="md-nav__link">
      Home
    </a>
  </li>

    
      
      
      


  <li class="md-nav__item">
    <a href="../../../trainers/trainers/" title="The Trainers" class="md-nav__link">
      The Trainers
    </a>
  </li>

    
      
      
      


  <li class="md-nav__item md-nav__item--nested">
    
      <input class="md-toggle md-nav__toggle" data-md-toggle="nav-3" type="checkbox" id="nav-3">
    
    <label class="md-nav__link" for="nav-3">
      Timetable
    </label>
    <nav class="md-nav" data-md-component="collapsible" data-md-level="1">
      <label class="md-nav__title" for="nav-3">
        Timetable
      </label>
      <ul class="md-nav__list" data-md-scrollfix>
        
        
          
          
          


  <li class="md-nav__item">
    <a href="../../../timetables/timetable_introNGS/" title="Intro to NGS" class="md-nav__link">
      Intro to NGS
    </a>
  </li>

        
      </ul>
    </nav>
  </li>

    
      
      
      


  <li class="md-nav__item">
    <a href="../../../preamble/" title="Workshop Information" class="md-nav__link">
      Workshop Information
    </a>
  </li>

    
      
      
      

  


  <li class="md-nav__item md-nav__item--active md-nav__item--nested">
    
      <input class="md-toggle md-nav__toggle" data-md-toggle="nav-5" type="checkbox" id="nav-5" checked>
    
    <label class="md-nav__link" for="nav-5">
      Introduction to NGS Modules
    </label>
    <nav class="md-nav" data-md-component="collapsible" data-md-level="1">
      <label class="md-nav__title" for="nav-5">
        Introduction to NGS Modules
      </label>
      <ul class="md-nav__list" data-md-scrollfix>
        
        
          
          
          


  <li class="md-nav__item">
    <a href="../../btp-module-ngs-cli/commandline/" title="Introduction to Command Line" class="md-nav__link">
      Introduction to Command Line
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../../btp-module-ngs-qc/ngs-qc/" title="Data Quality" class="md-nav__link">
      Data Quality
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../../btp-module-ngs-mapping/alignment/" title="Read Alignment" class="md-nav__link">
      Read Alignment
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../../btp-module-chip-seq/chip-seq/" title="ChIP-seq" class="md-nav__link">
      ChIP-seq
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../../btp-module-rna-seq/rna-seq/" title="RNA-Seq" class="md-nav__link">
      RNA-Seq
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../../btp-module-ngs-bio/biol/" title="NGS Biological Insight" class="md-nav__link">
      NGS Biological Insight
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../../btp-module-velvet/velvet/" title="de novo Genome Assembly" class="md-nav__link">
      de novo Genome Assembly
    </a>
  </li>

        
          
          
          

  


  <li class="md-nav__item md-nav__item--active">
    
    <input class="md-toggle md-nav__toggle" data-md-toggle="toc" type="checkbox" id="toc">
    
      
    
    
      <label class="md-nav__link md-nav__link--active" for="toc">
        DeNovo Canu
      </label>
    
    <a href="./" title="DeNovo Canu" class="md-nav__link md-nav__link--active">
      DeNovo Canu
    </a>
    
      
<nav class="md-nav md-nav--secondary">
  
  
    
  
  
    <label class="md-nav__title" for="toc">Table of contents</label>
    <ul class="md-nav__list" data-md-scrollfix>
      
        <li class="md-nav__item">
  <a href="#resources" title="Resources" class="md-nav__link">
    Resources
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#learning-objectives" title="Learning objectives" class="md-nav__link">
    Learning objectives
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#overview" title="Overview" class="md-nav__link">
    Overview
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#get-data" title="Get data" class="md-nav__link">
    Get data
  </a>
  
    <nav class="md-nav">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#ngs-workshop" title="NGS Workshop" class="md-nav__link">
    NGS Workshop
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#sample-information" title="Sample information" class="md-nav__link">
    Sample information
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
        <li class="md-nav__item">
  <a href="#assemble" title="Assemble" class="md-nav__link">
    Assemble
  </a>
  
    <nav class="md-nav">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#canu-output" title="Canu output" class="md-nav__link">
    Canu output
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#change-canu-parameters-if-required" title="Change Canu parameters if required" class="md-nav__link">
    Change Canu parameters if required
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#questions" title="Questions" class="md-nav__link">
    Questions
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
        <li class="md-nav__item">
  <a href="#trim-and-circularise" title="Trim and circularise" class="md-nav__link">
    Trim and circularise
  </a>
  
    <nav class="md-nav">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#run-circlator" title="Run Circlator" class="md-nav__link">
    Run Circlator
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#circlator-output" title="Circlator output" class="md-nav__link">
    Circlator output
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#options" title="Options" class="md-nav__link">
    Options
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#questions_1" title="Questions" class="md-nav__link">
    Questions
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
        <li class="md-nav__item">
  <a href="#find-smaller-plasmids" title="Find smaller plasmids" class="md-nav__link">
    Find smaller plasmids
  </a>
  
    <nav class="md-nav">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#align-illumina-reads-to-the-pacbio-contig" title="Align Illumina reads to the PacBio contig" class="md-nav__link">
    Align Illumina reads to the PacBio contig
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#extract-unmapped-illumina-reads" title="Extract unmapped Illumina reads" class="md-nav__link">
    Extract unmapped Illumina reads
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#assemble-the-unmapped-reads" title="Assemble the unmapped reads" class="md-nav__link">
    Assemble the unmapped reads
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#trim-the-plasmid" title="Trim the plasmid" class="md-nav__link">
    Trim the plasmid
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#plasmid-contig-orientation" title="Plasmid contig orientation" class="md-nav__link">
    Plasmid contig orientation
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#collect-contigs" title="Collect contigs" class="md-nav__link">
    Collect contigs
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#questions_2" title="Questions" class="md-nav__link">
    Questions
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
        <li class="md-nav__item">
  <a href="#correct" title="Correct" class="md-nav__link">
    Correct
  </a>
  
    <nav class="md-nav">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#make-an-alignment-file" title="Make an alignment file" class="md-nav__link">
    Make an alignment file
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#run-pilon" title="Run Pilon" class="md-nav__link">
    Run Pilon
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#genome-output" title="Genome output" class="md-nav__link">
    Genome output
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#questions_3" title="Questions" class="md-nav__link">
    Questions
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
        <li class="md-nav__item">
  <a href="#next" title="Next" class="md-nav__link">
    Next
  </a>
  
    <nav class="md-nav">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#further-analyses" title="Further analyses" class="md-nav__link">
    Further analyses
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#links" title="Links" class="md-nav__link">
    Links
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
      
      
    </ul>
  
</nav>
    
  </li>

        
      </ul>
    </nav>
  </li>

    
  </ul>
</nav>
                  </div>
                </div>
              </div>
            
            
              <div class="md-sidebar md-sidebar--secondary" data-md-component="toc">
                <div class="md-sidebar__scrollwrap">
                  <div class="md-sidebar__inner">
                    
<nav class="md-nav md-nav--secondary">
  
  
    
  
  
    <label class="md-nav__title" for="toc">Table of contents</label>
    <ul class="md-nav__list" data-md-scrollfix>
      
        <li class="md-nav__item">
  <a href="#resources" title="Resources" class="md-nav__link">
    Resources
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#learning-objectives" title="Learning objectives" class="md-nav__link">
    Learning objectives
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#overview" title="Overview" class="md-nav__link">
    Overview
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#get-data" title="Get data" class="md-nav__link">
    Get data
  </a>
  
    <nav class="md-nav">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#ngs-workshop" title="NGS Workshop" class="md-nav__link">
    NGS Workshop
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#sample-information" title="Sample information" class="md-nav__link">
    Sample information
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
        <li class="md-nav__item">
  <a href="#assemble" title="Assemble" class="md-nav__link">
    Assemble
  </a>
  
    <nav class="md-nav">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#canu-output" title="Canu output" class="md-nav__link">
    Canu output
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#change-canu-parameters-if-required" title="Change Canu parameters if required" class="md-nav__link">
    Change Canu parameters if required
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#questions" title="Questions" class="md-nav__link">
    Questions
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
        <li class="md-nav__item">
  <a href="#trim-and-circularise" title="Trim and circularise" class="md-nav__link">
    Trim and circularise
  </a>
  
    <nav class="md-nav">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#run-circlator" title="Run Circlator" class="md-nav__link">
    Run Circlator
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#circlator-output" title="Circlator output" class="md-nav__link">
    Circlator output
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#options" title="Options" class="md-nav__link">
    Options
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#questions_1" title="Questions" class="md-nav__link">
    Questions
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
        <li class="md-nav__item">
  <a href="#find-smaller-plasmids" title="Find smaller plasmids" class="md-nav__link">
    Find smaller plasmids
  </a>
  
    <nav class="md-nav">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#align-illumina-reads-to-the-pacbio-contig" title="Align Illumina reads to the PacBio contig" class="md-nav__link">
    Align Illumina reads to the PacBio contig
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#extract-unmapped-illumina-reads" title="Extract unmapped Illumina reads" class="md-nav__link">
    Extract unmapped Illumina reads
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#assemble-the-unmapped-reads" title="Assemble the unmapped reads" class="md-nav__link">
    Assemble the unmapped reads
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#trim-the-plasmid" title="Trim the plasmid" class="md-nav__link">
    Trim the plasmid
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#plasmid-contig-orientation" title="Plasmid contig orientation" class="md-nav__link">
    Plasmid contig orientation
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#collect-contigs" title="Collect contigs" class="md-nav__link">
    Collect contigs
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#questions_2" title="Questions" class="md-nav__link">
    Questions
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
        <li class="md-nav__item">
  <a href="#correct" title="Correct" class="md-nav__link">
    Correct
  </a>
  
    <nav class="md-nav">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#make-an-alignment-file" title="Make an alignment file" class="md-nav__link">
    Make an alignment file
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#run-pilon" title="Run Pilon" class="md-nav__link">
    Run Pilon
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#genome-output" title="Genome output" class="md-nav__link">
    Genome output
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#questions_3" title="Questions" class="md-nav__link">
    Questions
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
        <li class="md-nav__item">
  <a href="#next" title="Next" class="md-nav__link">
    Next
  </a>
  
    <nav class="md-nav">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#further-analyses" title="Further analyses" class="md-nav__link">
    Further analyses
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#links" title="Links" class="md-nav__link">
    Links
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
      
      
    </ul>
  
</nav>
                  </div>
                </div>
              </div>
            
          
          <div class="md-content">
            <article class="md-content__inner md-typeset">
              
                
                
                <h1 id="pacbio-reads-assembly-with-command-line-tools">Pacbio reads: assembly with command line tools<a class="headerlink" href="#pacbio-reads-assembly-with-command-line-tools" title="Permanent link">&para;</a></h1>
<!-- *Keywords: de novo assembly, PacBio, PacificBiosciences, Illumina, command line, Canu, Circlator, BWA, Spades, Pilon, Microbial Genomics Virtual Laboratory* -->

<p>This tutorial demonstrates how to use long PacBio sequence reads to assemble a bacterial genome, including correcting the assembly with short Illumina reads.</p>
<h2 id="resources">Resources<a class="headerlink" href="#resources" title="Permanent link">&para;</a></h2>
<p>Tools (and versions) used in this tutorial include:</p>
<ul>
<li>canu 1.5 (requires java 1.8)</li>
<li>infoseq and sizeseq (part of EMBOSS) 6.6.0.0</li>
<li>circlator 1.5.1</li>
<li>bwa 0.7.15</li>
<li>samtools 1.3.1</li>
<li>makeblastdb and blastn (part of blast) 2.4.0+</li>
<li>pilon 1.20</li>
<li>spades 3.10.1
<!-- - prokka 1.12 --></li>
</ul>
<h2 id="learning-objectives">Learning objectives<a class="headerlink" href="#learning-objectives" title="Permanent link">&para;</a></h2>
<p>At the end of this tutorial, be able to:</p>
<ol>
<li>Assemble and circularise a bacterial genome from PacBio sequence data.</li>
<li>Recover small plasmids missed by long read sequencing, using Illumina data</li>
<li>Explore the effect of polishing assembled sequences with a different data set.</li>
</ol>
<h2 id="overview">Overview<a class="headerlink" href="#overview" title="Permanent link">&para;</a></h2>
<p>Simplified version of workflow:</p>
<p><img alt="workflow" src="../images/flowchart.png" /></p>
<h2 id="get-data">Get data<a class="headerlink" href="#get-data" title="Permanent link">&para;</a></h2>
<p>The files we need are:</p>
<ul>
<li><strong>pacbio.fastq.gz</fn></strong> - the PacBio reads</li>
<li><strong><fn>illumina_R1.fastq.gz</fn></strong> - the Illumina forward reads</li>
<li><strong><fn>illumina_R2.fastq.gz</fn></strong> - the Illumina reverse reads</li>
</ul>
<!--

If you already have the files, skip forward to next section, [Assemble](#assemble).

Otherwise, this section has information about how to find and move the files:

### PacBio files


- Navigate to or create the directory in which you want to work.
- If the files are already on your server, you can symlink by using

wzxhzdk:0

- Alternatively, obtain the input files from elsewhere, e.g. from the BPA portal. (You will need a password.)

- Pacbio files are often stored in the format:
    - **<fn>Sample_name/Cell_name/Analysis_Results/long_file_name_1.fastq.gz</fn>**

- We will use the **<fn>longfilename.subreads.fastq.gz</fn>** files.

- The reads are usually split into three separate files because they are so large.

- Right click on the first **<fn>subreads.fastq.gz</fn>** file and "copy link address".

- In the command line, type:

wzxhzdk:1
- Repeat for the other two **<fn>subreads.fastq.gz</fn>** files.
- Join the files:
wzxhzdk:2
- If the files are not gzipped, type:
wzxhzdk:3

### Illumina files

- We will also use 2 x Illumina (Miseq) fastq.gz files.
- These are the **<fn>R1.fastq.gz</fn>** and **<fn>R2.fastq.gz</fn>** files.
- Symlink or "wget" these files as described above for PacBio files.
- Shorten the name of each of these files:

wzxhzdk:4
-->

<h3 id="ngs-workshop">NGS Workshop<a class="headerlink" href="#ngs-workshop" title="Permanent link">&para;</a></h3>
<p>Some of these tools will take too long to run in this workshop. For these tools, we have pre-computed the output files. In this workshop, we will still enter in the commands and set the tool running, but will sometimes then stop the run and move on to pre-computed output files.</p>
<p>In your directory, along with the PacBio and Illumina files, you may also see folders of pre-computed data.</p>
<!-- list of precomputed data: illuminaR1 and R2, pacbio.fastq.gz, canu_outdir, circlator_outdir, aln.bam, spades_assembly, aln_illumina_pacbio.bam, pilon1.fasta, pilon1.changes -->

<h3 id="sample-information">Sample information<a class="headerlink" href="#sample-information" title="Permanent link">&para;</a></h3>
<p>The sample used in this tutorial is a gram-positive bacteria called <em>Staphylococcus aureus</em> (sample number 25747). This particular sample is from a strain that is resistant to the antibiotic methicillin (a type of penicillin). It is also called MRSA: methicillin-resistant <em>Staphylococcus aureus</em>. It was isolated from (human) blood and caused bacteraemia, an infection of the bloodstream.</p>
<h2 id="assemble">Assemble<a name="assemble"></a><a class="headerlink" href="#assemble" title="Permanent link">&para;</a></h2>
<ul>
<li>We will use the assembly software called Canu, <a href="https://github.com/marbl/canu">https://github.com/marbl/canu</a>.</li>
<li>Run Canu with these commands:</li>
</ul>
<table class="codehilitetable"><tr><td class="linenos"><div class="linenodiv"><pre>1</pre></div></td><td class="code"><div class="codehilite"><pre><span></span>canu -p canu -d canu_outdir_NGS genomeSize=2.8m -pacbio-raw pacbio.fastq.gz
</pre></div>
</td></tr></table>

<ul>
<li>the first <code class="codehilite">canu</code> tells the program to run</li>
<li><code class="codehilite">-p canu</code> names prefix for output files (&ldquo;canu&rdquo;)</li>
<li><code class="codehilite">-d canu_outdir_NGS</code> names output directory</li>
<li>
<p><code class="codehilite">genomeSize</code> only has to be approximate.</p>
<ul>
<li>e.g. <em>Staphylococcus aureus</em>, 2.8m</li>
<li>e.g. <em>Streptococcus pyogenes</em>, 1.8m</li>
</ul>
</li>
<li>
<p>Canu will correct, trim and assemble the reads.</p>
</li>
<li>Various output will be displayed on the screen.</li>
</ul>
<p><em>NGS workshop: As we don&rsquo;t have time for Canu to complete, stop the run by typing <code class="codehilite">Ctrl-C</code>. We will look at pre-computed data in the folder <strong>canu_outdir</strong>.</em></p>
<h3 id="canu-output">Canu output<a class="headerlink" href="#canu-output" title="Permanent link">&para;</a></h3>
<p>Move into <strong><fn>canu_outdir</fn></strong> and <code class="codehilite">ls</code> to see the output files.</p>
<ul>
<li>The <strong><fn>canu.contigs.fasta</fn></strong> are the assembled sequences.</li>
<li>The <strong><fn>canu.unassembled.fasta</fn></strong> are the reads that could not be assembled.</li>
<li>The <strong><fn>canu.correctedReads.fasta.gz</fn></strong> are the corrected Pacbio reads that were used in the assembly.</li>
<li>The <strong><fn>canu.contigs.gfa</fn></strong> is the graph of the assembly.</li>
<li>Display summary information about the contigs: (<code class="codehilite">infoseq</code> is a tool from <a href="http://emboss.sourceforge.net/index.html">EMBOSS</a>)</li>
</ul>
<table class="codehilitetable"><tr><td class="linenos"><div class="linenodiv"><pre>1</pre></div></td><td class="code"><div class="codehilite"><pre><span></span>infoseq canu.contigs.fasta
</pre></div>
</td></tr></table>

<ul>
<li>This will show the contigs found by Canu. e.g.,</li>
</ul>
<table class="codehilitetable"><tr><td class="linenos"><div class="linenodiv"><pre>1</pre></div></td><td class="code"><div class="codehilite"><pre><span></span>    - tig00000001   2851805
</pre></div>
</td></tr></table>

<p>This looks like a chromosome of approximately 2.8 million bases.</p>
<p>This matches what we would expect for this sample. For other data, Canu may not be able to join all the reads into one contig, so there may be several contigs in the output. Also, the sample may contain some plasmids and these may be found full or partially by Canu as additional contigs.  </p>
<h3 id="change-canu-parameters-if-required">Change Canu parameters if required<a class="headerlink" href="#change-canu-parameters-if-required" title="Permanent link">&para;</a></h3>
<p>If the assembly is poor with many contigs, re-run Canu with extra sensitivity parameters; e.g.</p>
<table class="codehilitetable"><tr><td class="linenos"><div class="linenodiv"><pre>1</pre></div></td><td class="code"><div class="codehilite"><pre><span></span>canu -p prefix -d outdir corMhapSensitivity=high corMinCoverage=0 genomeSize=2.8m -pacbio-raw pacbio.fastq.gz
</pre></div>
</td></tr></table>

<h3 id="questions">Questions<a class="headerlink" href="#questions" title="Permanent link">&para;</a></h3>
<div class="admonition note">
<p class="admonition-title">Question</p>
<p>How do long- and short-read assembly methods differ?</p>
</div>
<div class="admonition success"><details><summary><strong>Answer</strong></summary><p>Short reads are usually assembled using De Bruijn graphs. With long reads, there is a move back towards simpler overlap-layout-consensus methods.</p></details></div>
<div class="admonition note">
<p class="admonition-title">Question</p>
<p>Where can we find out the what the approximate genome size should be for the species being assembled?</p>
</div>
<div class="admonition success"><details><summary><strong>Answer</strong></summary><p>Go to <a href="https://www.ncbi.nlm.nih.gov/genome/">https://www.ncbi.nlm.nih.gov/genome/</a> - enter species name - click on Genome Assembly and Annotation report - sort table by clicking on the column header Size (Mb) - look at range of sizes in this column.</p></details></div>
<div class="admonition note">
<p class="admonition-title">Question</p>
<p>Where could you view the output <strong>filename.gfa</strong> and what would it show?</p>
</div>
<div class="admonition success"><details><summary><strong>Answer</strong></summary><p>This is the assembly graph. You can view it using the tool &ldquo;Bandage&rdquo;, <a href="https://rrwick.github.io/Bandage/">https://rrwick.github.io/Bandage/</a>, to see how the contigs are connected (including ambiguities).</p></details></div>
<h2 id="trim-and-circularise">Trim and circularise<a class="headerlink" href="#trim-and-circularise" title="Permanent link">&para;</a></h2>
<h3 id="run-circlator">Run Circlator<a class="headerlink" href="#run-circlator" title="Permanent link">&para;</a></h3>
<p>Circlator (<a href="https://github.com/sanger-pathogens/circlator">https://github.com/sanger-pathogens/circlator</a>)
identifies and trims overhangs (on chromosomes and plasmids) and orients the start position at an appropriate gene (e.g. dnaA). It takes in the assembled contigs from Canu, as well as the corrected reads prepared by Canu.</p>
<p>Overhangs are shown in blue:</p>
<p><img alt="circlator" src="../images/circlator_diagram.png" />
<em>Adapted from Figure 1. Hunt et al. Genome Biology 2015</em></p>
<p>Move back into your main analysis folder.</p>
<p>Run Circlator:</p>
<table class="codehilitetable"><tr><td class="linenos"><div class="linenodiv"><pre>1</pre></div></td><td class="code"><div class="codehilite"><pre><span></span>circlator all --threads 4 --verbose canu_outdir/canu.contigs.fasta canu_outdir/canu.correctedReads.fasta.gz circlator_outdir_NGS
</pre></div>
</td></tr></table>

<ul>
<li><code class="codehilite">--threads</code> is the number of cores <!-- change this to an appropriate number--></li>
<li><code class="codehilite">--verbose</code> prints progress information to the screen</li>
<li><code class="codehilite">canu_outdir/canu.contigs.fasta</code> is the file path to the input Canu assembly</li>
<li><code class="codehilite">canu_outdir/canu.correctedReads.fasta.gz</code> is the file path to the corrected Pacbio reads - note, fastA not fastQ</li>
<li><code class="codehilite">circlator_outdir_NGS</code> is the name of the output directory.</li>
</ul>
<p><em>NGS workshop: Stop the run by typing <code class="codehilite">Ctrl-C</code>. We will look at pre-computed data in the folder <strong>circlator_outdir</strong>.</em></p>
<h3 id="circlator-output">Circlator output<a class="headerlink" href="#circlator-output" title="Permanent link">&para;</a></h3>
<p>Move into the <strong><fn>circlator_outdir</fn></strong> directory and <code class="codehilite">ls</code> to list files.</p>
<p><em>Were the contigs circularised?</em></p>
<table class="codehilitetable"><tr><td class="linenos"><div class="linenodiv"><pre>1</pre></div></td><td class="code"><div class="codehilite"><pre><span></span>less 04.merge.circularise.log
</pre></div>
</td></tr></table>

<ul>
<li>Yes, the contig was circularised (last column).</li>
<li>Type &ldquo;q&rdquo; to exit.</li>
</ul>
<p><em>Where were the contigs oriented (which gene)?</em></p>
<table class="codehilitetable"><tr><td class="linenos"><div class="linenodiv"><pre>1</pre></div></td><td class="code"><div class="codehilite"><pre><span></span>less 06.fixstart.log
</pre></div>
</td></tr></table>

<ul>
<li>Look in the &ldquo;gene_name&rdquo; column.</li>
<li>The contig has been oriented at tr|A0A090N2A8|A0A090N2A8_STAAU, which is another name for dnaA. <!-- (search swissprot - uniprot.org) --> This is typically used as the start of bacterial chromosome sequences.</li>
</ul>
<p><em>What are the trimmed contig sizes?</em></p>
<table class="codehilitetable"><tr><td class="linenos"><div class="linenodiv"><pre>1</pre></div></td><td class="code"><div class="codehilite"><pre><span></span>infoseq 06.fixstart.fasta
</pre></div>
</td></tr></table>

<ul>
<li>tig00000001 2823331 (28564 bases trimmed)</li>
</ul>
<p>This trimmed part is the overlap.</p>
<p><em>Re-name the contigs file</em>:</p>
<ul>
<li>The trimmed contigs are in the file called <strong><fn>06.fixstart.fasta</fn></strong>.</li>
<li>Re-name it <strong><fn>contig1.fasta</fn></strong>:</li>
</ul>
<table class="codehilitetable"><tr><td class="linenos"><div class="linenodiv"><pre>1</pre></div></td><td class="code"><div class="codehilite"><pre><span></span>cp 06.fixstart.fasta contig1.fasta
</pre></div>
</td></tr></table>

<p>Open this file in a text editor (e.g. nano: <code class="codehilite">nano contig1.fasta</code>) and change the header to &ldquo;&gt;chromosome&rdquo;.</p>
<p>Move the file back into the main folder (<code class="codehilite">mv contig1.fasta ../</code>).</p>
<h3 id="options">Options<a class="headerlink" href="#options" title="Permanent link">&para;</a></h3>
<p>If all the contigs have not circularised with Circlator, an option is to change the <code class="codehilite">--b2r_length_cutoff</code> setting to approximately 2X the average read depth.</p>
<h3 id="questions_1">Questions<a class="headerlink" href="#questions_1" title="Permanent link">&para;</a></h3>
<div class="admonition note">
<p class="admonition-title">Question</p>
<p>Were all the contigs circularised? Why/why not?</p>
</div>
<div class="admonition success"><details><summary><strong>Answer</strong></summary><p>In this example, the contig could be circularized because it contained the entire sequence, with overhangs that were trimmed.</p></details></div>
<div class="admonition note">
<p class="admonition-title">Question</p>
<p>Circlator can set the start of the sequence at a particular gene. Which gene does it use? Is this appropriate for all contigs?</p>
</div>
<div class="admonition success"><details><summary><strong>Answer</strong></summary><p>Circlator uses dnaA for the chromosomal contig. For other contigs, it uses a centrally-located gene. However, ideally, plasmids would be oriented on a gene such as a rep gene. It is possible to provide a file to Circlator to do this.</p></details></div>
<h2 id="find-smaller-plasmids">Find smaller plasmids<a class="headerlink" href="#find-smaller-plasmids" title="Permanent link">&para;</a></h2>
<p>Pacbio reads are long, and may have been longer than small plasmids. We will look for any small plasmids using the Illumina reads.</p>
<p>This section involves several steps:</p>
<ol>
<li>Use the Canu+Circlator output of a trimmed assembly contig.</li>
<li>Map all the Illumina reads against this PacBio-assembled contig.</li>
<li>Extract any reads that <em>didn&rsquo;t</em> map and assemble them together: this could be a plasmid, or part of a plasmid.</li>
<li>Look for overhang: if found, trim.</li>
</ol>
<h3 id="align-illumina-reads-to-the-pacbio-contig">Align Illumina reads to the PacBio contig<a class="headerlink" href="#align-illumina-reads-to-the-pacbio-contig" title="Permanent link">&para;</a></h3>
<ul>
<li>Index the contigs file:</li>
</ul>
<table class="codehilitetable"><tr><td class="linenos"><div class="linenodiv"><pre>1</pre></div></td><td class="code"><div class="codehilite"><pre><span></span>bwa index contig1.fasta
</pre></div>
</td></tr></table>

<ul>
<li>Align Illumina reads using using bwa mem:</li>
</ul>
<table class="codehilitetable"><tr><td class="linenos"><div class="linenodiv"><pre>1</pre></div></td><td class="code"><div class="codehilite"><pre><span></span>bwa mem -t 4 contig1.fasta illumina_R1.fastq.gz illumina_R2.fastq.gz | samtools sort &gt; aln_NGS.bam
</pre></div>
</td></tr></table>

<ul>
<li><code class="codehilite">bwa mem</code> is the alignment tool</li>
<li><code class="codehilite">-t 4</code> is the number of cores <!-- choose an appropriate number --></li>
<li><code class="codehilite">contig1.fasta</code> is the input assembly file</li>
<li><code class="codehilite">illumina_R1.fastq.gz illumina_R2.fastq.gz</code> are the Illumina reads</li>
<li><code class="codehilite">| samtools sort</code> pipes the output to samtools to sort</li>
<li><code class="codehilite">&gt; aln_NGS.bam</code> sends the alignment to the file <strong><fn>aln_NGS.bam</fn></strong></li>
</ul>
<p><em>NGS workshop: Stop the run by typing <code class="codehilite">Ctrl-C</code>. We will use the pre-computed file called <strong>aln.bam</strong>.</em></p>
<h3 id="extract-unmapped-illumina-reads">Extract unmapped Illumina reads<a class="headerlink" href="#extract-unmapped-illumina-reads" title="Permanent link">&para;</a></h3>
<ul>
<li>Index the alignment file:</li>
</ul>
<table class="codehilitetable"><tr><td class="linenos"><div class="linenodiv"><pre>1</pre></div></td><td class="code"><div class="codehilite"><pre><span></span>samtools index aln.bam
</pre></div>
</td></tr></table>

<ul>
<li>Extract the fastq files from the bam alignment - those reads that were unmapped to the Pacbio alignment - and save them in various &ldquo;unmapped&rdquo; files:</li>
</ul>
<table class="codehilitetable"><tr><td class="linenos"><div class="linenodiv"><pre>1</pre></div></td><td class="code"><div class="codehilite"><pre><span></span>samtools fastq -f 4 -1 unmapped.R1.fastq -2 unmapped.R2.fastq -s unmapped.RS.fastq aln.bam
</pre></div>
</td></tr></table>

<ul>
<li><code class="codehilite">fastq</code> is a command that coverts a <strong><fn>.bam</fn></strong> file into fastq format</li>
<li><code class="codehilite">-f 4</code> : only output unmapped reads</li>
<li><code class="codehilite">-1</code> : put R1 reads into a file called <strong><fn>unmapped.R1.fastq</fn></strong></li>
<li><code class="codehilite">-2</code> : put R2 reads into a file called <strong><fn>unmapped.R2.fastq</fn></strong></li>
<li><code class="codehilite">-s</code> : put singleton reads into a file called <strong><fn>unmapped.RS.fastq</fn></strong></li>
<li><code class="codehilite">aln.bam</code> : input alignment file</li>
</ul>
<p>We now have three files of the unampped reads: <strong><fn> unmapped.R1.fastq</fn></strong>, <strong><fn> unmapped.R2.fastq</fn></strong>, <strong><fn> unmapped.RS.fastq</fn></strong>.</p>
<h3 id="assemble-the-unmapped-reads">Assemble the unmapped reads<a class="headerlink" href="#assemble-the-unmapped-reads" title="Permanent link">&para;</a></h3>
<ul>
<li>Assemble with Spades (<a href="http://cab.spbu.ru/software/spades/">http://cab.spbu.ru/software/spades/</a>):</li>
</ul>
<table class="codehilitetable"><tr><td class="linenos"><div class="linenodiv"><pre>1</pre></div></td><td class="code"><div class="codehilite"><pre><span></span>spades.py -1 unmapped.R1.fastq -2 unmapped.R2.fastq -s unmapped.RS.fastq --careful --cov-cutoff auto -o spades_assembly_NGS
</pre></div>
</td></tr></table>

<ul>
<li><code class="codehilite">-1</code> is input file forward</li>
<li><code class="codehilite">-2</code> is input file reverse</li>
<li><code class="codehilite">-s</code> is unpaired</li>
<li><code class="codehilite">--careful</code> minimizes mismatches and short indels</li>
<li><code class="codehilite">--cov-cutoff auto</code> computes the coverage threshold (rather than the default setting, &ldquo;off&rdquo;)</li>
<li><code class="codehilite">-o</code> is the output directory</li>
</ul>
<p><em>NGS workshop: Stop the run by typing <code class="codehilite">Ctrl-C</code>. We will use the pre-computed file in the folder <strong>spades_assembly</strong>.</em></p>
<p>Move into the output directory (<strong><fn>spades_assembly</fn></strong>) and look at the contigs:</p>
<table class="codehilitetable"><tr><td class="linenos"><div class="linenodiv"><pre>1</pre></div></td><td class="code"><div class="codehilite"><pre><span></span>infoseq contigs.fasta
</pre></div>
</td></tr></table>

<ul>
<li>78 contigs were assembled, with the max length of 2250 (the first contig).  </li>
<li>All other nodes are &lt; 650kb so we will disregard as they are unlikely to be plasmids.</li>
<li>We will extract the first sequence (NODE_1):</li>
</ul>
<table class="codehilitetable"><tr><td class="linenos"><div class="linenodiv"><pre>1</pre></div></td><td class="code"><div class="codehilite"><pre><span></span>samtools faidx contigs.fasta
</pre></div>
</td></tr></table>

<table class="codehilitetable"><tr><td class="linenos"><div class="linenodiv"><pre>1</pre></div></td><td class="code"><div class="codehilite"><pre><span></span>samtools faidx contigs.fasta NODE_1_length_2550_cov_496.613 &gt; contig2.fasta
</pre></div>
</td></tr></table>

<ul>
<li>This is now saved as <strong><fn>contig2.fasta</fn></strong></li>
<li>Open in nano and change header to &ldquo;&gt;plasmid&rdquo;.</li>
</ul>
<h3 id="trim-the-plasmid">Trim the plasmid<a class="headerlink" href="#trim-the-plasmid" title="Permanent link">&para;</a></h3>
<p>To trim any overhang on this plasmid, we will blast the start of contig2 against itself.</p>
<ul>
<li>Take the start of the contig:</li>
</ul>
<table class="codehilitetable"><tr><td class="linenos"><div class="linenodiv"><pre>1</pre></div></td><td class="code"><div class="codehilite"><pre><span></span>head -n 10 contig2.fasta &gt; contig2.fa.head
</pre></div>
</td></tr></table>

<ul>
<li>We want to see if it matches the end (overhang).</li>
<li>Format the assembly file for blast:</li>
</ul>
<table class="codehilitetable"><tr><td class="linenos"><div class="linenodiv"><pre>1</pre></div></td><td class="code"><div class="codehilite"><pre><span></span>makeblastdb -in contig2.fasta -dbtype nucl
</pre></div>
</td></tr></table>

<ul>
<li>Blast the start of the assembly (.head file) against all of the assembly:</li>
</ul>
<table class="codehilitetable"><tr><td class="linenos"><div class="linenodiv"><pre>1</pre></div></td><td class="code"><div class="codehilite"><pre><span></span>blastn -query contig2.fa.head -db contig2.fasta -evalue 1e-3 -dust no -out contig2.bls
</pre></div>
</td></tr></table>

<ul>
<li>Look at <strong><fn>contig2.bls</fn></strong> to see hits:</li>
</ul>
<table class="codehilitetable"><tr><td class="linenos"><div class="linenodiv"><pre>1</pre></div></td><td class="code"><div class="codehilite"><pre><span></span>less contig2.bls
</pre></div>
</td></tr></table>

<ul>
<li>The first hit is at start, as expected.</li>
<li>The second hit is at 2474 all the way to the end - 2550.</li>
<li>This is the overhang.</li>
<li>Trim to position 2473.</li>
<li>Type &lsquo;q&rsquo; to exit.</li>
<li>Index the plasmid.fa file:</li>
</ul>
<table class="codehilitetable"><tr><td class="linenos"><div class="linenodiv"><pre>1</pre></div></td><td class="code"><div class="codehilite"><pre><span></span>samtools faidx contig2.fasta
</pre></div>
</td></tr></table>

<ul>
<li>Trim</li>
</ul>
<table class="codehilitetable"><tr><td class="linenos"><div class="linenodiv"><pre>1</pre></div></td><td class="code"><div class="codehilite"><pre><span></span>samtools faidx contig2.fasta plasmid:1-2473 &gt; plasmid.fa.trimmed
</pre></div>
</td></tr></table>

<ul>
<li>
<p><code class="codehilite">plasmid</code> is the name of the contig, and we want the sequence from 1-2473.</p>
</li>
<li>
<p>Open this file in nano (<code class="codehilite">nano plasmid.fa.trimmed</code>) and change the header to &ldquo;&gt;plasmid&rdquo;, save.</p>
</li>
<li>(Use the side scroll bar to see the top of the file.)</li>
<li>We now have a trimmed plasmid.</li>
<li>Move file back into main folder:</li>
</ul>
<table class="codehilitetable"><tr><td class="linenos"><div class="linenodiv"><pre>1</pre></div></td><td class="code"><div class="codehilite"><pre><span></span>cp plasmid.fa.trimmed ../
</pre></div>
</td></tr></table>

<ul>
<li>Move into the main folder.</li>
</ul>
<h3 id="plasmid-contig-orientation">Plasmid contig orientation<a class="headerlink" href="#plasmid-contig-orientation" title="Permanent link">&para;</a></h3>
<p>The bacterial chromosome was oriented at the gene dnaA. Plasmids are often oriented at the replication gene, but this is highly variable and there is no established convention. Here we will orient the plasmid at a gene found by Prodigal, in Circlator:</p>
<table class="codehilitetable"><tr><td class="linenos"><div class="linenodiv"><pre>1</pre></div></td><td class="code"><div class="codehilite"><pre><span></span>circlator fixstart plasmid.fa.trimmed plasmid_fixstart
</pre></div>
</td></tr></table>

<ul>
<li><code class="codehilite">fixstart</code> is an option in Circlator just to orient a sequence.</li>
<li><code class="codehilite">plasmid.fa.trimmed</code> is our small plasmid.</li>
<li><code class="codehilite">plasmid_fixstart</code> is the prefix for the output files.</li>
</ul>
<p>View the output:</p>
<table class="codehilitetable"><tr><td class="linenos"><div class="linenodiv"><pre>1</pre></div></td><td class="code"><div class="codehilite"><pre><span></span>less plasmid_fixstart.log
</pre></div>
</td></tr></table>

<ul>
<li>The plasmid has been oriented at a gene predicted by Prodigal, and the break-point is at position 1200.</li>
<li>Change the file name:</li>
</ul>
<table class="codehilitetable"><tr><td class="linenos"><div class="linenodiv"><pre>1</pre></div></td><td class="code"><div class="codehilite"><pre><span></span>cp plasmid_fixstart.fasta contig2.fasta
</pre></div>
</td></tr></table>

<!-- note: annotated with prokka. plasmid only has two proteins. ermC, and a hypothetical protein. protein blast genbank: matches a staph replication and maintenance protein. -->

<h3 id="collect-contigs">Collect contigs<a class="headerlink" href="#collect-contigs" title="Permanent link">&para;</a></h3>
<table class="codehilitetable"><tr><td class="linenos"><div class="linenodiv"><pre>1</pre></div></td><td class="code"><div class="codehilite"><pre><span></span>cat contig1.fasta contig2.fasta &gt; genome.fasta
</pre></div>
</td></tr></table>

<ul>
<li>See the contigs and sizes:</li>
</ul>
<table class="codehilitetable"><tr><td class="linenos"><div class="linenodiv"><pre>1</pre></div></td><td class="code"><div class="codehilite"><pre><span></span>infoseq genome.fasta
</pre></div>
</td></tr></table>

<ul>
<li>chromosome: 2823331</li>
<li>plasmid: 2473</li>
</ul>
<h3 id="questions_2">Questions<a class="headerlink" href="#questions_2" title="Permanent link">&para;</a></h3>
<div class="admonition note">
<p class="admonition-title">Question</p>
<p>Why is this section so complicated?</p>
</div>
<div class="admonition success"><details><summary><strong>Answer</strong></summary><p>Finding small plasmids is difficult for many reasons! This paper has a nice summary: On the (im)possibility to reconstruct plasmids from whole genome short-read sequencing data. doi: <a href="https://doi.org/10.1101/086744">https://doi.org/10.1101/086744</a></p></details></div>
<div class="admonition note">
<p class="admonition-title">Question</p>
<p>Why can PacBio sequencing miss small plasmids?</p>
</div>
<div class="admonition success"><details><summary><strong>Answer</strong></summary><p>Library prep size selection</p></details></div>
<div class="admonition note">
<p class="admonition-title">Question</p>
<p>We extract unmapped Illumina reads and assemble these to find small plasmids. What could they be missing?</p>
</div>
<div class="admonition success"><details><summary><strong>Answer</strong></summary><p>Repeats that have mapped to the PacBio assembly.</p></details></div>
<div class="admonition note">
<p class="admonition-title">Question</p>
<p>How do you find a plasmid in a Bandage graph?</p>
</div>
<div class="admonition success"><details><summary><strong>Answer</strong></summary><p>It is probably circular, matches the size of a known plasmid, and has a rep gene.</p></details></div>
<div class="admonition note">
<p class="admonition-title">Question</p>
<p>Are there easier ways to find plasmids?</p>
</div>
<div class="admonition success"><details><summary><strong>Answer</strong></summary><p>Possibly. One option is the program called Unicycler which may automate many of these steps. <a href="https://github.com/rrwick/Unicycler">https://github.com/rrwick/Unicycler</a></p></details></div>
<h2 id="correct">Correct<a class="headerlink" href="#correct" title="Permanent link">&para;</a></h2>
<p>We will correct the Pacbio assembly with Illumina reads, using the tool Pilon (<a href="https://github.com/broadinstitute/pilon/wiki">https://github.com/broadinstitute/pilon/wiki</a>).</p>
<h3 id="make-an-alignment-file">Make an alignment file<a class="headerlink" href="#make-an-alignment-file" title="Permanent link">&para;</a></h3>
<ul>
<li>Align the Illumina reads (R1 and R2) to the draft PacBio assembly, e.g. <strong><fn>genome.fasta</fn></strong>:</li>
</ul>
<table class="codehilitetable"><tr><td class="linenos"><div class="linenodiv"><pre>1
2</pre></div></td><td class="code"><div class="codehilite"><pre><span></span>bwa index genome.fasta
bwa mem -t 4 genome.fasta illumina_R1.fastq.gz illumina_R2.fastq.gz | samtools sort &gt; aln_illumina_pacbio_NGS.bam
</pre></div>
</td></tr></table>

<ul>
<li><code class="codehilite">-t</code> is the number of cores <!-- set this to an appropriate number. (To find out how many you have, <code class="codehilite">wzxhzdk:86</code>) --></li>
</ul>
<p><em>NGS workshop: Stop the run by typing <code class="codehilite">Ctrl-C</code>. We will use the pre-computed file called <strong>aln_illumina_pacbio.bam</strong>.</em></p>
<ul>
<li>Index the files:</li>
</ul>
<table class="codehilitetable"><tr><td class="linenos"><div class="linenodiv"><pre>1
2</pre></div></td><td class="code"><div class="codehilite"><pre><span></span>samtools index aln_illumina_pacbio.bam
samtools faidx genome.fasta
</pre></div>
</td></tr></table>

<ul>
<li>Now we have an alignment file to use in Pilon: <strong><fn>aln_illumina_pacbio.bam</fn></strong></li>
</ul>
<h3 id="run-pilon">Run Pilon<a class="headerlink" href="#run-pilon" title="Permanent link">&para;</a></h3>
<ul>
<li>Run:</li>
</ul>
<table class="codehilitetable"><tr><td class="linenos"><div class="linenodiv"><pre>1</pre></div></td><td class="code"><div class="codehilite"><pre><span></span>pilon --genome genome.fasta --frags aln_illumina_pacbio.bam --output pilon1_NGS --fix all --mindepth 0.5 --changes --verbose --threads 4
</pre></div>
</td></tr></table>

<ul>
<li><code class="codehilite">--genome</code> is the name of the input assembly to be corrected</li>
<li><code class="codehilite">--frags</code> is the alignment of the reads against the assembly</li>
<li><code class="codehilite">--output</code> is the name of the output prefix</li>
<li><code class="codehilite">--fix</code> is an option for types of corrections</li>
<li><code class="codehilite">--mindepth</code> gives a minimum read depth to use</li>
<li><code class="codehilite">--changes</code> produces an output file of the changes made</li>
<li><code class="codehilite">--verbose</code> prints information to the screen during the run</li>
<li><code class="codehilite">--threads</code>: the number of cores</li>
</ul>
<p><em>NGS workshop: Stop the run by typing <code class="codehilite">Ctrl-C</code>. We will use the pre-computed files called with the prefixes <strong>pilon1</strong>.</em></p>
<p>Look at the changes file:</p>
<table class="codehilitetable"><tr><td class="linenos"><div class="linenodiv"><pre>1</pre></div></td><td class="code"><div class="codehilite"><pre><span></span>less pilon1.changes
</pre></div>
</td></tr></table>

<p><em>Example:</em></p>
<p><img alt="pilon" src="../images/pilon.png" /></p>
<p>Look at the details of the fasta file:</p>
<table class="codehilitetable"><tr><td class="linenos"><div class="linenodiv"><pre>1</pre></div></td><td class="code"><div class="codehilite"><pre><span></span>infoseq pilon1.fasta
</pre></div>
</td></tr></table>

<ul>
<li>chromosome - 2823340 (net +9 bases)</li>
<li>plasmid - 2473 (no change)</li>
</ul>
<p><strong>Option:</strong></p>
<p>If there are many changes, run Pilon again, using the <strong><fn>pilon1.fasta</fn></strong> file as the input assembly, and the Illumina reads to correct.</p>
<h3 id="genome-output">Genome output<a class="headerlink" href="#genome-output" title="Permanent link">&para;</a></h3>
<ul>
<li>Change the file name:</li>
</ul>
<table class="codehilitetable"><tr><td class="linenos"><div class="linenodiv"><pre>1</pre></div></td><td class="code"><div class="codehilite"><pre><span></span>cp pilon1.fasta assembly.fasta
</pre></div>
</td></tr></table>

<ul>
<li>We now have the corrected genome assembly of <em>Staphylococcus aureus</em> in .fasta format, containing a chromosome and a small plasmid.  </li>
</ul>
<h3 id="questions_3">Questions<a class="headerlink" href="#questions_3" title="Permanent link">&para;</a></h3>
<div class="admonition note">
<p class="admonition-title">Question</p>
<p>Why don&rsquo;t we correct earlier in the assembly process?</p>
</div>
<div class="admonition success"><details><summary><strong>Answer</strong></summary><p>We need to circularise the contigs and trim overhangs first.</p></details></div>
<div class="admonition note">
<p class="admonition-title">Question</p>
<p>Why can we use some reads (Illumina) to correct other reads (PacBio) ?</p>
</div>
<div class="admonition success"><details><summary><strong>Answer</strong></summary><p>Illumina reads have higher accuracy</p></details></div>
<div class="admonition note">
<p class="admonition-title">Question</p>
<p>Could we just use PacBio reads to assemble the genome?</p>
</div>
<div class="admonition success"><details><summary><strong>Answer</strong></summary><p>Yes, if accuracy adequate.</p></details></div>
<!-- ## Short-read assembly: a comparison

So far, we have assembled the long PacBio reads into one contig (the chromosome) and found an additional plasmid in the Illumina short reads.

If we only had Illumina reads, we could also assemble these using the tool Spades.

You can try this here or try it later on your own data.

### Get data

We will use the same Illumina data as we used above:

- **<fn>illumina_R1.fastq.gz</fn>**: the Illumina forward reads
- **<fn>illumina_R2.fastq.gz</fn>**: the Illumina reverse reads

### Assemble

Run Spades:

wzxhzdk:40

- `-1` is input file of forward reads
- `-2` is input file of reverse reads
- `--careful` minimizes mismatches and short indels
- `--cov-cutoff auto` computes the coverage threshold (rather than the default setting, "off")
- `-o` is the output directory

### Results

Move into the output directory and look at the contigs:

wzxhzdk:41

### Questions

How many contigs were found by Spades?

- many

How does this compare to the number of contigs found by assembling the long read data with Canu?

- many more.

Does it matter that an assembly is in many contigs?

- Yes

  - broken genes => missing/incorrect annotations
  - less information about structure: e.g. number of plasmids

- No

  - Many or all genes may still be annotated
  - Gene location is useful (e.g. chromosome, plasmid1) but not always essential (e.g. presence/absence of particular resistance genes)

How can we get more information about the assembly from Spades?

- Look at the assembly graph **<fn>assembly_graph.fastg</fn>**, e.g. in the program Bandage. This shows how contigs are related, albeit with ambiguity in some places.

-->

<h2 id="next">Next<a class="headerlink" href="#next" title="Permanent link">&para;</a></h2>
<h3 id="further-analyses">Further analyses<a class="headerlink" href="#further-analyses" title="Permanent link">&para;</a></h3>
<ul>
<li>Annotate genomes, e.g. with Prokka, <a href="https://github.com/tseemann/prokka">https://github.com/tseemann/prokka</a></li>
<li>Comparative genomics, e.g. with Roary, <a href="https://sanger-pathogens.github.io/Roary/">https://sanger-pathogens.github.io/Roary/</a></li>
</ul>
<h3 id="links">Links<a class="headerlink" href="#links" title="Permanent link">&para;</a></h3>
<!--- [Details of bas.h5 files](https://s3.amazonaws.com/files.pacb.com/software/instrument/2.0.0/bas.h5+Reference+Guide.pdf)
-->

<ul>
<li>Canu <a href="http://canu.readthedocs.io/en/stable/quick-start.html">manual</a> and <a href="https://github.com/marbl/canu">gitub repository</a></li>
<li>Circlator <a href="http://genomebiology.biomedcentral.com/articles/10.1186/s13059-015-0849-0">article</a> and <a href="http://sanger-pathogens.github.io/circlator/">github repository</a></li>
<li>Pilon <a href="http://journals.plos.org/plosone/article?id=10.1371/journal.pone.0112963">article</a> and <a href="https://github.com/broadinstitute/pilon/wiki">github repository</a></li>
<li>Notes on <a href="https://github.com/PacificBiosciences/Bioinformatics-Training/wiki/Finishing-Bacterial-Genomes">finishing</a> and <a href="https://github.com/PacificBiosciences/Bioinformatics-Training/wiki/Evaluating-Assemblies">evaluating</a> assemblies.</li>
</ul>
                
                  
                
              
              
                
              
            </article>
          </div>
        </div>
      </main>
      
        
<footer class="md-footer">
  
    <div class="md-footer-nav">
      <nav class="md-footer-nav__inner md-grid">
        
          <a href="../../btp-module-velvet/velvet/" title="de novo Genome Assembly" class="md-flex md-footer-nav__link md-footer-nav__link--prev" rel="prev">
            <div class="md-flex__cell md-flex__cell--shrink">
              <i class="md-icon md-icon--arrow-back md-footer-nav__button"></i>
            </div>
            <div class="md-flex__cell md-flex__cell--stretch md-footer-nav__title">
              <span class="md-flex__ellipsis">
                <span class="md-footer-nav__direction">
                  Previous
                </span>
                de novo Genome Assembly
              </span>
            </div>
          </a>
        
        
      </nav>
    </div>
  
  <div class="md-footer-meta md-typeset">
    <div class="md-footer-meta__inner md-grid">
      <div class="md-footer-copyright">
        
          <div class="md-footer-copyright__highlight">
            &copy; The authors; 2017-
          </div>
        
        powered by
        <a href="http://www.mkdocs.org" title="MkDocs">MkDocs</a>
        and
        <a href="http://squidfunk.github.io/mkdocs-material/" title="Material for MkDocs">
          Material for MkDocs</a>
      </div>
      
        
      
    </div>
  </div>
</footer>
      
    </div>
    
      <script src="../../../assets/javascripts/application-c35428f87f.js"></script>
      
      
      <script>app.initialize({url:{base:"../../.."}})</script>
      
        <script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.1/MathJax.js?config=TeX-AMS-MML_HTMLorMML"></script>
      
        <script src="../../../js/details.js"></script>
      
    
    
      
    
  </body>
</html>