
<!DOCTYPE html>
<html lang="en" class="no-js">
  <head>
    
      <meta charset="utf-8">
      <meta name="viewport" content="width=device-width,initial-scale=1">
      
        <meta name="description" content="This contains all material for Markdown version of BTP workshops.">
      
      
      
        <meta name="author" content="Authors">
      
      
        <link rel="shortcut icon" href="../../../assets/images/favicon.png">
      
      <meta name="generator" content="mkdocs-0.16.3, mkdocs-material-1.7.3">
    
    
      
        <title>Assessing somatic mutational signatures - Markdown Manuals for Bioinformatics Training Platform (BTP)</title>
      
    
    
      <script src="../../../assets/javascripts/modernizr-1df76c4e58.js"></script>
    
    
      <link rel="stylesheet" href="../../../assets/stylesheets/application-acc56dad49.css">
      
        <link rel="stylesheet" href="../../../assets/stylesheets/application-8817cfa535.palette.css">
      
    
    
      
        
        
        
        <link rel="stylesheet" href="https://fonts.googleapis.com/css?family=Roboto:300,400,400i,700|Roboto+Mono">
        <style>body,input{font-family:"Roboto","Helvetica Neue",Helvetica,Arial,sans-serif}code,kbd,pre{font-family:"Roboto Mono","Courier New",Courier,monospace}</style>
      
      <link rel="stylesheet" href="https://fonts.googleapis.com/icon?family=Material+Icons">
    
    
      <link rel="stylesheet" href="../../../css/details.css">
    
    
  </head>
  
  
  
  
    <body data-md-color-primary="indigo" data-md-color-accent="light-blue">
  
    <svg class="md-svg">
      <defs>
        
        
      </defs>
    </svg>
    <input class="md-toggle" data-md-toggle="drawer" type="checkbox" id="drawer">
    <input class="md-toggle" data-md-toggle="search" type="checkbox" id="search">
    <label class="md-overlay" data-md-component="overlay" for="drawer"></label>
    
      <header class="md-header" data-md-component="header">
  <nav class="md-header-nav md-grid">
    <div class="md-flex">
      <div class="md-flex__cell md-flex__cell--shrink">
        
          <a href="../../.." title="Markdown Manuals for Bioinformatics Training Platform (BTP)" class="md-logo md-header-nav__button">
            <img src="../../../media/logos/bpalogo.png" width="24" height="24">
          </a>
        
      </div>
      <div class="md-flex__cell md-flex__cell--shrink">
        <label class="md-icon md-icon--menu md-header-nav__button" for="drawer"></label>
      </div>
      <div class="md-flex__cell md-flex__cell--stretch">
        <span class="md-flex__ellipsis md-header-nav__title">
          
            
              
                <span class="md-header-nav__parent">
                  Cancer Genomics Modules
                </span>
              
                <span class="md-header-nav__parent">
                  Somatic Mutations
                </span>
              
            
            Assessing somatic mutational signatures
          
        </span>
      </div>
      <div class="md-flex__cell md-flex__cell--shrink">
        
          <label class="md-icon md-icon--search md-header-nav__button" for="search"></label>
          
<div class="md-search" data-md-component="search">
  <label class="md-search__overlay" for="search"></label>
  <div class="md-search__inner">
    <form class="md-search__form" name="search">
      <input type="text" class="md-search__input" name="query" required placeholder="Search" accesskey="s" autocapitalize="off" autocorrect="off" autocomplete="off" spellcheck="false" data-md-component="query">
      <label class="md-icon md-search__icon" for="search"></label>
      <button type="reset" class="md-icon md-search__icon" data-md-component="reset">close</button>
    </form>
    <div class="md-search__output">
      <div class="md-search__scrollwrap" data-md-scrollfix>
        <div class="md-search-result" data-md-component="result" data-md-lang-search="">
          <div class="md-search-result__meta" data-md-lang-result-none="No matching documents" data-md-lang-result-one="1 matching document" data-md-lang-result-other="# matching documents">
            Type to start searching
          </div>
          <ol class="md-search-result__list"></ol>
        </div>
      </div>
    </div>
  </div>
</div>
        
      </div>
      
    </div>
  </nav>
</header>
    
    <div class="md-container">
      
      
        

  

<nav class="md-tabs md-tabs--active" data-md-component="tabs">
  <div class="md-tabs__inner md-grid">
    <ul class="md-tabs__list">
      
        
  <li class="md-tabs__item">
    
      <a href="../../.." title="Home" class="md-tabs__link">
        Home
      </a>
    
  </li>

      
        
  
  
    <li class="md-tabs__item">
      
        <a href="../../../timetables/timetable_introNGS/" title="Timetable" class="md-tabs__link">
          Timetable
        </a>
      
    </li>
  

      
        
      
        
  
  
    <li class="md-tabs__item">
      
        <a href="../../btp-module-chip-seq/chip-seq/" title="BTP Modules" class="md-tabs__link">
          BTP Modules
        </a>
      
    </li>
  

      
        
  
  
    <li class="md-tabs__item">
      
        <a href="../../cancer-module-alignment/alignment/" title="Cancer Genomics Modules" class="md-tabs__link md-tabs__link--active">
          Cancer Genomics Modules
        </a>
      
    </li>
  

      
        
  
  
    <li class="md-tabs__item">
      
        <a href="../../denovo-module-cli/commandline/" title="DeNovo Modules" class="md-tabs__link">
          DeNovo Modules
        </a>
      
    </li>
  

      
        
  
  
    <li class="md-tabs__item">
      
        <a href="../../metagenomics-module-qc/ngs-qc/" title="Metagenomics Modules" class="md-tabs__link">
          Metagenomics Modules
        </a>
      
    </li>
  

      
    </ul>
  </div>
</nav>
      
      <main class="md-main">
        <div class="md-main__inner md-grid" data-md-component="container">
          
            
              <div class="md-sidebar md-sidebar--primary" data-md-component="navigation">
                <div class="md-sidebar__scrollwrap">
                  <div class="md-sidebar__inner">
                    <nav class="md-nav md-nav--primary" data-md-level="0">
  <label class="md-nav__title md-nav__title--site" for="drawer">
    
      <i class="md-logo md-nav__button">
        <img src="../../../media/logos/bpalogo.png">
      </i>
    
    Markdown Manuals for Bioinformatics Training Platform (BTP)
  </label>
  
  <ul class="md-nav__list" data-md-scrollfix>
    
      
      
      


  <li class="md-nav__item">
    <a href="../../.." title="Home" class="md-nav__link">
      Home
    </a>
  </li>

    
      
      
      


  <li class="md-nav__item md-nav__item--nested">
    
      <input class="md-toggle md-nav__toggle" data-md-toggle="nav-2" type="checkbox" id="nav-2">
    
    <label class="md-nav__link" for="nav-2">
      Timetable
    </label>
    <nav class="md-nav" data-md-component="collapsible" data-md-level="1">
      <label class="md-nav__title" for="nav-2">
        Timetable
      </label>
      <ul class="md-nav__list" data-md-scrollfix>
        
        
          
          
          


  <li class="md-nav__item">
    <a href="../../../timetables/timetable_introNGS/" title="Intro to NGS" class="md-nav__link">
      Intro to NGS
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../../../timetables/timetable_cancer/" title="Cancer Genomics" class="md-nav__link">
      Cancer Genomics
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../../../timetables/timetable_longRead/" title="Long Read" class="md-nav__link">
      Long Read
    </a>
  </li>

        
      </ul>
    </nav>
  </li>

    
      
      
      


  <li class="md-nav__item">
    <a href="../../../preamble/" title="Workshop Information" class="md-nav__link">
      Workshop Information
    </a>
  </li>

    
      
      
      


  <li class="md-nav__item md-nav__item--nested">
    
      <input class="md-toggle md-nav__toggle" data-md-toggle="nav-4" type="checkbox" id="nav-4">
    
    <label class="md-nav__link" for="nav-4">
      BTP Modules
    </label>
    <nav class="md-nav" data-md-component="collapsible" data-md-level="1">
      <label class="md-nav__title" for="nav-4">
        BTP Modules
      </label>
      <ul class="md-nav__list" data-md-scrollfix>
        
        
          
          
          


  <li class="md-nav__item">
    <a href="../../btp-module-chip-seq/chip-seq/" title="ChIP-seq" class="md-nav__link">
      ChIP-seq
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../../btp-module-denovo-canu/denovo_canu/" title="DeNovo Canu" class="md-nav__link">
      DeNovo Canu
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../../btp-module-ngs-cli/commandline/" title="Introduction to Command Line" class="md-nav__link">
      Introduction to Command Line
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../../btp-module-ngs-qc/ngs-qc/" title="Data Quality" class="md-nav__link">
      Data Quality
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../../btp-module-ngs-mapping/alignment/" title="Read Alignment" class="md-nav__link">
      Read Alignment
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../../btp-module-rna-seq/rna-seq/" title="RNA-Seq" class="md-nav__link">
      RNA-Seq
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../../btp-module-velvet/velvet/" title="de novo Genome Assembly" class="md-nav__link">
      de novo Genome Assembly
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../../btp-module-ngs-bio/biol/" title="NGS Biological Insight" class="md-nav__link">
      NGS Biological Insight
    </a>
  </li>

        
      </ul>
    </nav>
  </li>

    
      
      
      

  


  <li class="md-nav__item md-nav__item--active md-nav__item--nested">
    
      <input class="md-toggle md-nav__toggle" data-md-toggle="nav-5" type="checkbox" id="nav-5" checked>
    
    <label class="md-nav__link" for="nav-5">
      Cancer Genomics Modules
    </label>
    <nav class="md-nav" data-md-component="collapsible" data-md-level="1">
      <label class="md-nav__title" for="nav-5">
        Cancer Genomics Modules
      </label>
      <ul class="md-nav__list" data-md-scrollfix>
        
        
          
          
          


  <li class="md-nav__item">
    <a href="../../cancer-module-alignment/alignment/" title="Read Alignment" class="md-nav__link">
      Read Alignment
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../../cancer-module-snv/snv/" title="Single Nucleotide Variant Calling and Annotation" class="md-nav__link">
      Single Nucleotide Variant Calling and Annotation
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../../cancer-module-cnv/cnv-tut/" title="Copy Number Variation" class="md-nav__link">
      Copy Number Variation
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../../cancer-module-sv/sv_tut/" title="Structural Variant Analysis" class="md-nav__link">
      Structural Variant Analysis
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../../cancer-module-viz/visu/" title="Variant Visualisation" class="md-nav__link">
      Variant Visualisation
    </a>
  </li>

        
          
          
          

  


  <li class="md-nav__item md-nav__item--active md-nav__item--nested">
    
      <input class="md-toggle md-nav__toggle" data-md-toggle="nav-5-6" type="checkbox" id="nav-5-6" checked>
    
    <label class="md-nav__link" for="nav-5-6">
      Somatic Mutations
    </label>
    <nav class="md-nav" data-md-component="collapsible" data-md-level="2">
      <label class="md-nav__title" for="nav-5-6">
        Somatic Mutations
      </label>
      <ul class="md-nav__list" data-md-scrollfix>
        
        
          
          
          

  


  <li class="md-nav__item md-nav__item--active">
    
    <input class="md-toggle md-nav__toggle" data-md-toggle="toc" type="checkbox" id="toc">
    
    
      <label class="md-nav__link md-nav__link--active" for="toc">
        Assessing somatic mutational signatures
      </label>
    
    <a href="./" title="Assessing somatic mutational signatures" class="md-nav__link md-nav__link--active">
      Assessing somatic mutational signatures
    </a>
    
      
<nav class="md-nav md-nav--secondary">
  
  
  
    <label class="md-nav__title" for="toc">Table of contents</label>
    <ul class="md-nav__list" data-md-scrollfix>
      
        <li class="md-nav__item">
  <a href="#key-learning-outcomes" title="Key Learning Outcomes" class="md-nav__link">
    Key Learning Outcomes
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#resources-youll-be-using" title="Resources You’ll be Using" class="md-nav__link">
    Resources You’ll be Using
  </a>
  
    <nav class="md-nav">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#tools-used" title="Tools Used" class="md-nav__link">
    Tools Used
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#sources-of-data" title="Sources of Data" class="md-nav__link">
    Sources of Data
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#useful-links" title="Useful Links" class="md-nav__link">
    Useful Links
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
        <li class="md-nav__item">
  <a href="#introduction" title="Introduction" class="md-nav__link">
    Introduction
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#preparing-the-r-environment" title="Preparing the R environment" class="md-nav__link">
    Preparing the R environment
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#loading-and-preparing-the-snv-mutation-data" title="Loading and preparing the SNV mutation data" class="md-nav__link">
    Loading and preparing the SNV mutation data
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#snv-mutation-context" title="SNV mutation context" class="md-nav__link">
    SNV mutation context
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#running-the-nmf-analysis" title="Running the NMF analysis" class="md-nav__link">
    Running the NMF analysis
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#interpreting-the-signature-results" title="Interpreting the signature results" class="md-nav__link">
    Interpreting the signature results
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#references" title="References" class="md-nav__link">
    References
  </a>
  
</li>
      
      
      
    </ul>
  
</nav>
    
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../02_intogen/" title="Cohort analysis for the identification of driver genes" class="md-nav__link">
      Cohort analysis for the identification of driver genes
    </a>
  </li>

        
      </ul>
    </nav>
  </li>

        
      </ul>
    </nav>
  </li>

    
      
      
      


  <li class="md-nav__item md-nav__item--nested">
    
      <input class="md-toggle md-nav__toggle" data-md-toggle="nav-6" type="checkbox" id="nav-6">
    
    <label class="md-nav__link" for="nav-6">
      DeNovo Modules
    </label>
    <nav class="md-nav" data-md-component="collapsible" data-md-level="1">
      <label class="md-nav__title" for="nav-6">
        DeNovo Modules
      </label>
      <ul class="md-nav__list" data-md-scrollfix>
        
        
          
          
          


  <li class="md-nav__item">
    <a href="../../denovo-module-cli/commandline/" title="DeNovo Command Line" class="md-nav__link">
      DeNovo Command Line
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../../denovo-module-bac/bac_asm/" title="Bacterial Assembly" class="md-nav__link">
      Bacterial Assembly
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../../denovo-module-euk/eukaryotic_assem/" title="Eukaryote Assembly" class="md-nav__link">
      Eukaryote Assembly
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../../denovo-module-scaffolding/scaffold/" title="Scaffolding" class="md-nav__link">
      Scaffolding
    </a>
  </li>

        
      </ul>
    </nav>
  </li>

    
      
      
      


  <li class="md-nav__item md-nav__item--nested">
    
      <input class="md-toggle md-nav__toggle" data-md-toggle="nav-7" type="checkbox" id="nav-7">
    
    <label class="md-nav__link" for="nav-7">
      Metagenomics Modules
    </label>
    <nav class="md-nav" data-md-component="collapsible" data-md-level="1">
      <label class="md-nav__title" for="nav-7">
        Metagenomics Modules
      </label>
      <ul class="md-nav__list" data-md-scrollfix>
        
        
          
          
          


  <li class="md-nav__item">
    <a href="../../metagenomics-module-qc/ngs-qc/" title="Metagenomics QC" class="md-nav__link">
      Metagenomics QC
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../../metagenomics-module-tax/tax/" title="Metagenomics Taxonomic" class="md-nav__link">
      Metagenomics Taxonomic
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../../metagenomics-module-vis/vis/" title="Metagenomics Visualisation" class="md-nav__link">
      Metagenomics Visualisation
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../../metagenomics-module-wga/wga/" title="Metagenomics WGA" class="md-nav__link">
      Metagenomics WGA
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../../metagenomics-module-fda/fda/" title="Metagenomics Functional Analysis" class="md-nav__link">
      Metagenomics Functional Analysis
    </a>
  </li>

        
      </ul>
    </nav>
  </li>

    
  </ul>
</nav>
                  </div>
                </div>
              </div>
            
            
              <div class="md-sidebar md-sidebar--secondary" data-md-component="toc">
                <div class="md-sidebar__scrollwrap">
                  <div class="md-sidebar__inner">
                    
<nav class="md-nav md-nav--secondary">
  
  
  
    <label class="md-nav__title" for="toc">Table of contents</label>
    <ul class="md-nav__list" data-md-scrollfix>
      
        <li class="md-nav__item">
  <a href="#key-learning-outcomes" title="Key Learning Outcomes" class="md-nav__link">
    Key Learning Outcomes
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#resources-youll-be-using" title="Resources You’ll be Using" class="md-nav__link">
    Resources You’ll be Using
  </a>
  
    <nav class="md-nav">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#tools-used" title="Tools Used" class="md-nav__link">
    Tools Used
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#sources-of-data" title="Sources of Data" class="md-nav__link">
    Sources of Data
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#useful-links" title="Useful Links" class="md-nav__link">
    Useful Links
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
        <li class="md-nav__item">
  <a href="#introduction" title="Introduction" class="md-nav__link">
    Introduction
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#preparing-the-r-environment" title="Preparing the R environment" class="md-nav__link">
    Preparing the R environment
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#loading-and-preparing-the-snv-mutation-data" title="Loading and preparing the SNV mutation data" class="md-nav__link">
    Loading and preparing the SNV mutation data
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#snv-mutation-context" title="SNV mutation context" class="md-nav__link">
    SNV mutation context
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#running-the-nmf-analysis" title="Running the NMF analysis" class="md-nav__link">
    Running the NMF analysis
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#interpreting-the-signature-results" title="Interpreting the signature results" class="md-nav__link">
    Interpreting the signature results
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#references" title="References" class="md-nav__link">
    References
  </a>
  
</li>
      
      
      
    </ul>
  
</nav>
                  </div>
                </div>
              </div>
            
          
          <div class="md-content">
            <article class="md-content__inner md-typeset">
              
                
                
                  <h1>Assessing somatic mutational signatures</h1>
                
                <h2 id="key-learning-outcomes">Key Learning Outcomes<a class="headerlink" href="#key-learning-outcomes" title="Permanent link">&para;</a></h2>
<p>After completing this practical the trainee should be able to:</p>
<ul>
<li>
<p>Visualise mutational signatures present in a cohort using somatic
    single nucleotide mutation data in Variant Call Format (VCF) files.</p>
</li>
<li>
<p>Compare analysis output with published results to identify common
    mutational signatures.</p>
</li>
<li>
<p>Have gained overview knowledge of how somatic signatures can help
    with cohort cancer analysis.</p>
</li>
</ul>
<hr />
<h2 id="resources-youll-be-using">Resources You’ll be Using<a class="headerlink" href="#resources-youll-be-using" title="Permanent link">&para;</a></h2>
<h3 id="tools-used">Tools Used<a class="headerlink" href="#tools-used" title="Permanent link">&para;</a></h3>
<p>R-3.2.2 statistical environment:<br />
<a href="https://www.r-project.org/">https://www.r-project.org/</a></p>
<p>SomaticSignatures R package:<br />
<a href="http://bioconductor.org/packages/release/bioc/html/SomaticSignatures.html">http://bioconductor.org/packages/release/bioc/html/SomaticSignatures.html</a></p>
<p>BSgenome.Hsapiens.UCSC.hg19:<br />
<a href="http://bioconductor.org/packages/release/data/annotation/html/BSgenome.Hsapiens.UCSC.hg19.html">http://bioconductor.org/packages/release/data/annotation/html/BSgenome.Hsapiens.UCSC.hg19.html</a></p>
<p>VariantAnnotation:<br />
<a href="https://bioconductor.org/packages/release/bioc/html/VariantAnnotation.html">https://bioconductor.org/packages/release/bioc/html/VariantAnnotation.html</a></p>
<p>GenomicRanges:<br />
<a href="https://bioconductor.org/packages/release/bioc/html/GenomicRanges.html">https://bioconductor.org/packages/release/bioc/html/GenomicRanges.html</a></p>
<p>Cairo:<br />
<a href="https://cran.rstudio.com/web/packages/Cairo/index.html">https://cran.rstudio.com/web/packages/Cairo/index.html</a></p>
<h3 id="sources-of-data">Sources of Data<a class="headerlink" href="#sources-of-data" title="Permanent link">&para;</a></h3>
<p>TCGA melanoma SNV data:<br />
<a href="https://tcga-data.nci.nih.gov/tcga/">https://tcga-data.nci.nih.gov/tcga/</a></p>
<p>ICGC ovarian SNV data: <br />
<a href="https://dcc.icgc.org/">https://dcc.icgc.org/</a></p>
<h3 id="useful-links">Useful Links<a class="headerlink" href="#useful-links" title="Permanent link">&para;</a></h3>
<p>Variant Call Format (VCF) specification: <br />
<a href="http://samtools.github.io/hts-specs/VCFv4.2.pdf">http://samtools.github.io/hts-specs/VCFv4.2.pdf</a></p>
<hr />
<h2 id="introduction">Introduction<a class="headerlink" href="#introduction" title="Permanent link">&para;</a></h2>
<p>The most common genetic model for cancer development is the accumulation
of DNA mutations over time, eventually leading to the disruption or
dysregulation of enough key genes that lead cells to uncontrolled
growth. Cells in our bodies accumulate DNA mutations over time due to
normal aging processes and through exposure to carcinogens.</p>
<p>Recently researchers found a method to take all the single nucleotide
mutations identified in tumour cells (somatic SNVs) and group them
together by the type of the mutation and also what the neighbouring
bases are. This is commonly referred to as somatic mutational
signatures. Common mutational processes that are regularly identified in cancer
sequencing are:</p>
<ul>
<li>
<p>Age: the aging process. These are high in C/T transitions due to
    deamination of methyl-cytidine.</p>
</li>
<li>
<p>Smoking: marks exposure to inhaled carcinogens and has high numbers
    of C/A transversions.</p>
</li>
<li>
<p>UV: UV exposure. These are also high in C/T transitions at
    di-pyrimidine sites.</p>
</li>
<li>
<p>BRCA: Indicates that the homologous recombination repair pathway is
    defective.</p>
</li>
<li>
<p>APOBEC: Thought to be marking dysregulated APOBEC enzyme activity on
    single stranded DNA produced during the repair processing of other
    lesions such as double stand breaks.</p>
</li>
<li>
<p>MMR: Mismatch repair pathway not working properly. These are high in
    C/T mutations too.</p>
</li>
</ul>
<p><br>
In cohort cancer analysis it is common to try to generate subtypes to
group your data based on a particular molecular phenotype. A reason for
doing may include finding sets of patients that have a similar form of
the disease and therefore all might benefit from a particular treatment.
We can use the somatic mutational signatures analysis to group the data
from a cohort of patients to inform which genomes are most similar based
on the pattern of exposures or processes that have contributed to their
genome changes. The patients don’t have to have the same type of cancer
so pan-cancer studies are using this analysis to find similarities
across cancer types.</p>
<hr />
<h2 id="preparing-the-r-environment">Preparing the R environment<a class="headerlink" href="#preparing-the-r-environment" title="Permanent link">&para;</a></h2>
<p>The mathematical framework developed by Alexandrov <em>et al.</em> was implemented
in MATLAB. We are going to use a version implemented in R by Gehring <em>et
al.</em> called <code class="codehilite">SomaticSignatures package</code>, that is very quick and flexible
but currently only accepts point mutations not insertions or deletions
(indels). In tests on our data we have found that the Somatic Signatures
package in R returns very similar results to the full implementation of
Alexandrov’s framework.</p>
<p>The data files you will need are contained in the subdirectory called
<code class="codehilite">somatic/somatic_signatures</code>:</p>
<p>Open the Terminal and go to the <code class="codehilite">somatic_signatures</code> working directory:</p>
<table class="codehilitetable"><tr><td class="linenos"><div class="linenodiv"><pre>1
2</pre></div></td><td class="code"><div class="codehilite"><pre><span></span><span class="nb">cd</span> ~/somatic/somatic_signatures
<span class="nb">pwd</span>
</pre></div>
</td></tr></table>

<p>In this folder you should find 12 files that end with the extension
<code class="codehilite"><span class="na">.vcf</span></code>. Use the list command to make sure you can see them.</p>
<table class="codehilitetable"><tr><td class="linenos"><div class="linenodiv"><pre>1</pre></div></td><td class="code"><div class="codehilite"><pre><span></span>ls
</pre></div>
</td></tr></table>

<p>These files contain data extracted from the TCGA melanoma paper and
Australian ICGC ovarian paper both mentioned in the introductory slides.
They have been edited in order to allow this practical to run quickly
and are not good examples of VCF files.</p>
<p>Start R and set the working directory. Just start by typing R onto the
command line.</p>
<table class="codehilitetable"><tr><td class="linenos"><div class="linenodiv"><pre>1</pre></div></td><td class="code"><div class="codehilite"><pre><span></span>R
</pre></div>
</td></tr></table>

<p>Load all the package libraries needed for this analysis by running the
commands.</p>
<table class="codehilitetable"><tr><td class="linenos"><div class="linenodiv"><pre>1
2
3
4</pre></div></td><td class="code"><div class="codehilite"><pre><span></span><span class="kn">library</span><span class="p">(</span>SomaticSignatures<span class="p">)</span>
<span class="kn">library</span><span class="p">(</span>BSgenome.Hsapiens.UCSC.hg19<span class="p">)</span>
<span class="kn">library</span><span class="p">(</span>ggplot2<span class="p">)</span>
<span class="kn">library</span><span class="p">(</span>Cairo<span class="p">)</span>
</pre></div>
</td></tr></table>

<p>Set the directory where any output files will be generated</p>
<table class="codehilitetable"><tr><td class="linenos"><div class="linenodiv"><pre>1</pre></div></td><td class="code"><div class="codehilite"><pre><span></span><span class="kp">setwd</span><span class="p">(</span><span class="s">&quot;~/somatic/somatic_signatures&quot;</span><span class="p">)</span>
</pre></div>
</td></tr></table>

<hr />
<h2 id="loading-and-preparing-the-snv-mutation-data">Loading and preparing the SNV mutation data<a class="headerlink" href="#loading-and-preparing-the-snv-mutation-data" title="Permanent link">&para;</a></h2>
<p>The mutations used in this analysis need to be high quality somatic
mutations</p>
<ul>
<li>
<p>Remember the goal is to find the key mutational processes that these
    tumours have been exposed to, so you need to exclude germline
    mutations (mutations that the person was born with that can be seen
    in the sequencing of matched normal samples).</p>
</li>
<li>
<p>Sequencing errors can also occur at particular DNA sequence contexts
    and can also be picked up using this method. To avoid this use only
    high quality mutation calls.</p>
</li>
</ul>
<p>Read in the mutations from the 12 VCF files</p>
<table class="codehilitetable"><tr><td class="linenos"><div class="linenodiv"><pre>1</pre></div></td><td class="code"><div class="codehilite"><pre><span></span>files <span class="o">&lt;-</span> <span class="kp">list.files</span><span class="p">(</span><span class="s">&quot;~/somatic/somatic_signatures&quot;</span><span class="p">,</span> pattern<span class="o">=</span><span class="s">&quot;vcf$&quot;</span><span class="p">,</span> full.names<span class="o">=</span><span class="kc">TRUE</span><span class="p">)</span>
</pre></div>
</td></tr></table>

<p>To make sure all the files are listed run the command.</p>
<table class="codehilitetable"><tr><td class="linenos"><div class="linenodiv"><pre>1</pre></div></td><td class="code"><div class="codehilite"><pre><span></span>files
</pre></div>
</td></tr></table>

<p>You should see a list of 12 sample files.</p>
<p>Next read in all the genomic positions of variants in the VCF files
using the <code class="codehilite">vranges</code> class.</p>
<table class="codehilitetable"><tr><td class="linenos"><div class="linenodiv"><pre>1</pre></div></td><td class="code"><div class="codehilite"><pre><span></span>vranges <span class="o">&lt;-</span> <span class="kp">lapply</span><span class="p">(</span>files<span class="p">,</span> <span class="kr">function</span><span class="p">(</span>v<span class="p">)</span> readVcfAsVRanges<span class="p">(</span>v<span class="p">,</span><span class="s">&quot;hg19&quot;</span><span class="p">))</span>
</pre></div>
</td></tr></table>

<p>Join all the lists of variant positions into one big data set so that it
can be processed together and look at what is contained in the
concatenated <code class="codehilite">vranges</code> data</p>
<table class="codehilitetable"><tr><td class="linenos"><div class="linenodiv"><pre>1
2</pre></div></td><td class="code"><div class="codehilite"><pre><span></span>vranges.cat <span class="o">&lt;-</span> <span class="kp">do.call</span><span class="p">(</span><span class="kt">c</span><span class="p">,</span>vranges<span class="p">)</span>
vranges.cat
</pre></div>
</td></tr></table>

<p>The first line of output of the <code class="codehilite">vranges.cat</code> shows us that in total we
have put over 100,000 mutations recording the chromosome positions and
mutation base changes along with what sample they were seen in.</p>
<p>Note there are a lot of NA values in this data set because we have left
out non-essential information in order to cut down on the processing
time.</p>
<p>Next we need to ensure all the positions in the <code class="codehilite">vranges</code> object have been
recorded in UCSC notation form so that they will match up with the
reference we are using.</p>
<table class="codehilitetable"><tr><td class="linenos"><div class="linenodiv"><pre>1</pre></div></td><td class="code"><div class="codehilite"><pre><span></span>vranges.cat <span class="o">&lt;-</span> ucsc<span class="p">(</span>vranges.cat<span class="p">)</span>
</pre></div>
</td></tr></table>

<p>It is always important to select the correct reference for your data.</p>
<p>We can print out how many mutations we have read in for each of the
cancer samples we are using by using the command.</p>
<table class="codehilitetable"><tr><td class="linenos"><div class="linenodiv"><pre>1</pre></div></td><td class="code"><div class="codehilite"><pre><span></span><span class="kp">print</span><span class="p">(</span><span class="kp">table</span><span class="p">(</span>sampleNames<span class="p">(</span>vranges.cat<span class="p">)))</span>
</pre></div>
</td></tr></table>

<p>We have now added all the positional and base change information now we
can use the reference and the position of the mutation to look up the
bases on either side of the mutation i.e. the mutation context.</p>
<p>Run the mutationContext function of SomaticSignatures.</p>
<table class="codehilitetable"><tr><td class="linenos"><div class="linenodiv"><pre>1</pre></div></td><td class="code"><div class="codehilite"><pre><span></span>mc <span class="o">&lt;-</span> mutationContext<span class="p">(</span>vranges.cat<span class="p">,</span> BSgenome.Hsapiens.UCSC.hg19<span class="p">)</span>
</pre></div>
</td></tr></table>

<p>We can inspect what information we had added to the <code class="codehilite">vranges.cat</code> object
by typing <code class="codehilite">mc</code> on the command line. Notice that the mutation and its
context have been added to the last two columns.</p>
<table class="codehilitetable"><tr><td class="linenos"><div class="linenodiv"><pre>1</pre></div></td><td class="code"><div class="codehilite"><pre><span></span>mc
</pre></div>
</td></tr></table>

<hr />
<h2 id="snv-mutation-context">SNV mutation context<a class="headerlink" href="#snv-mutation-context" title="Permanent link">&para;</a></h2>
<p>There are a total of 96 possible single base mutations and context
combinations. We can calculate this by listing out the six possible
types of single nucleotide mutations:</p>
<p><pre>
  -   C/A   the reverse compliment (G/T) is also in this group
  -   C/G   includes (G/C)
  -   C/T   includes (G/C)
  -   T/A   includes (A/T)
  -   T/C   includes (A/G)
  -   T/G   includes (A/C)
  </pre></p>
<p>The neighbouring bases, on either side of a mutation, are referred to as
the mutation context. There are 16 possible combinations of mutation
contexts. Here [.] stands for one of the mutations listed above.</p>
<p><pre>
  -   A[.]A   A[.]C   A[.]G   A[.]T
  -   C[.]A   C[.]C   C[.]G   C[.]T
  -   G[.]A   G[.]C   G[.]G   G[.]T
  -   T[.]A   T[.]C   T[.]G   T[.]T
  </pre></p>
<p>Now if we substitute the [.]’s with each of the 6 different mutations
you will find there are 96 possible types of combined mutations and
contexts (6 x 16).</p>
<p>Start by substituting [.] for the A/C mutation type</p>
<p><pre>
  -   A[C/A]A
  -   A[C/A]C
  -   A[C/A]G
  -   A[C/A]T
  -   C[C/A]A
  -   C[C/A]C
  -   C[C/A]G
  </pre></p>
<p>and so on&hellip;</p>
<p>We assign all the somatic mutations identified in a single tumour to one
of these categories and total up the number in each.</p>
<div class="admonition note">
<p class="admonition-title">Question</p>
<p>What about a mutation that looks like G[A/C]A, where should this go?</p>
<div class="admonition hint"><details><summary><strong>Hint</strong></summary><p>Remember to reverse compliment all the nucleotides.</p></details></div>
<div class="admonition success"><details><summary><strong>Answer</strong></summary><p>In the T[A/C]C context count.</p></details></div>
</div>
<p><br>
Now we have all the information that is needed for each sample we can
make a matrix that contains counts of mutations in each of the 96
possible combinations of mutations and contexts counting up the totals
separately for each sample</p>
<table class="codehilitetable"><tr><td class="linenos"><div class="linenodiv"><pre>1
2</pre></div></td><td class="code"><div class="codehilite"><pre><span></span>mm <span class="o">&lt;-</span> motifMatrix<span class="p">(</span>mc<span class="p">,</span> group <span class="o">=</span> <span class="s">&quot;sampleNames&quot;</span><span class="p">,</span> normalize<span class="o">=</span><span class="kc">TRUE</span><span class="p">)</span>
<span class="kp">dim</span><span class="p">(</span>mm<span class="p">)</span>
</pre></div>
</td></tr></table>

<p>The output of the <code class="codehilite">dim(mm)</code> command show us that there are 96 rows
(these are the context values) and 12 columns which are the 12 samples.</p>
<hr />
<h2 id="running-the-nmf-analysis">Running the NMF analysis<a class="headerlink" href="#running-the-nmf-analysis" title="Permanent link">&para;</a></h2>
<p>Using the matrix we have made we can now run the non-negative matrix
factorisation (NMF) process that attempts to find the most stable,
grouping solutions for all of the combinations of mutations and
contexts. It does this by trying to find similar patterns, or profiles,
amongst the samples to sort the data into firstly just 2 groups. This is
repeated to get replicate values for each attempt and then separating
the data by 3 groups, and then 4 and so on.</p>
<p>These parameter choices have been made to keep running time short for
this practical. If you have more samples from potentially diverse
sources you may need to run with a larger range of signatures and with
more replicates.</p>
<p>To find out how many signatures we have in the data run the command.</p>
<table class="codehilitetable"><tr><td class="linenos"><div class="linenodiv"><pre>1</pre></div></td><td class="code"><div class="codehilite"><pre><span></span>gof_nmf <span class="o">&lt;-</span> assessNumberSignatures<span class="p">(</span>mm<span class="p">,</span> <span class="m">2</span><span class="o">:</span><span class="m">10</span><span class="p">,</span> nReplicates <span class="o">=</span> <span class="m">5</span><span class="p">)</span>
</pre></div>
</td></tr></table>

<p>Visualise the results from the NMF processing by making a pdf of the
plot</p>
<table class="codehilitetable"><tr><td class="linenos"><div class="linenodiv"><pre>1
2
3</pre></div></td><td class="code"><div class="codehilite"><pre><span></span>Cairo<span class="p">(</span>file<span class="o">=</span><span class="s">&quot;plotNumberOfSignatures.pdf&quot;</span><span class="p">,</span> type<span class="o">=</span><span class="s">&quot;pdf&quot;</span><span class="p">,</span> units<span class="o">=</span><span class="s">&quot;in&quot;</span><span class="p">,</span> width<span class="o">=</span><span class="m">9</span><span class="p">,</span> height<span class="o">=</span><span class="m">8</span><span class="p">,</span> dpi<span class="o">=</span><span class="m">72</span><span class="p">)</span>
plotNumberSignatures<span class="p">(</span>gof_nmf<span class="p">)</span>
dev.off<span class="p">()</span>
</pre></div>
</td></tr></table>

<p>Open up the PDF and examine the curve. The plotNumberOfSignatures PDF
that will have been made in the working directory that you set up at the
beginning</p>
<p><img alt="image" src="../images/numberofsignatures.png" /> <strong>Figure 1:</strong> This plot is used to find the number of signatures that is likely to be the best grouping solution. The top plot shows the decreasing residual sum of squares for each
increasing number of signatures and the bottom plot the increasing explained variance as
the number of potential signatures increases. Ideally the best solution will be the lowest
number of signatures with a low RSS and a high explained variance.</p>
<p><br>
Look at the y-axis scale on the bottom panel of Figure 1. The explained
variance is already very high and so close to finding the correct
solution for the number of signatures even with just 2. The error bars
around each point are fairly small considering we have a very small
sample set. Deciding how many signatures are present can be tricky but
here let’s go for 3. This is where the gradient of both curves have
started to flatten out.</p>
<p>Now run the NMF again but this time stipulating that you want to group
the data into 3 different mutational signatures.</p>
<table class="codehilitetable"><tr><td class="linenos"><div class="linenodiv"><pre>1</pre></div></td><td class="code"><div class="codehilite"><pre><span></span>sigs_nmf = identifySignatures(mm, 3, nmfDecomposition)
</pre></div>
</td></tr></table>

<p>Visualise the shape of the profiles for these 3 signatures</p>
<table class="codehilitetable"><tr><td class="linenos"><div class="linenodiv"><pre>1
2
3</pre></div></td><td class="code"><div class="codehilite"><pre><span></span>Cairo(file=&quot;plot3Signatures.pdf&quot;, type=&quot;pdf&quot;, units=&quot;in&quot;, width=10, height=8, dpi=72)
plotSignatures(sigs_nmf,normalize=TRUE, percent=FALSE) + ggtitle(&quot;Somatic Signatures: NMF - Barchart&quot;) + scale_fill_brewer(palette = &quot;Set2&quot;)
dev.off()
</pre></div>
</td></tr></table>

<p>Open up <code class="codehilite">plot3Signatures.pdf</code> that will have been made in the working
directory.</p>
<p>You should have generated a plot with three signature profiles obtained
from the NMF processing of the test dataset.</p>
<p>The 96 possible mutation/context combinations are plotted along the x
axis arranged in blocks of 6 lots of 16 (see information above). The
height of the bars indicates the frequency of those particular mutation
and context combinations in each signature.</p>
<p>Although the section colours are different to the plot you have
generated the mutations are still in the same order across the plot.</p>
<hr />
<h2 id="interpreting-the-signature-results">Interpreting the signature results<a class="headerlink" href="#interpreting-the-signature-results" title="Permanent link">&para;</a></h2>
<p>In their paper Alexandrov <em>et al.</em> used this analysis to generate profiles
from the data for more than 7000 tumour samples sequenced through both
exome and whole genome approaches. They were able to group the data to
reveal which genomes have been exposed to similar mutational processes
contributing to the genome mutations.</p>
<p><img alt="image" src="../images/alexandrov_nature.png" /> <strong>Figure 2.</strong> The 96 possible mutation/context
combinations are plotted along the x axis arranged in blocks of 6 lots of 16.
The y axis indicates the frequency of those particular mutation and context
combinations in each signature. <em>Source</em>: Alexandrov <em>et al.</em> Nature 2013</p>
<p><br>  </p>
<div class="admonition note">
<p class="admonition-title">Question</p>
<p>Can you match up, by eye, the profile shapes against a selection of
known mutational signatures supplied (Figure 2)?</p>
<p>Try to match up the patterns made by the positions of the highest peaks for each signature.</p>
<div class="admonition success"><details><summary><strong>Answer</strong></summary><ul><li><p>Alexandrov signature 7 matches with our signature 1</p></li><li><p>Alexandrov signature 13 matches with our signature 2</p></li><li><p>Alexandrov signature 3 matches with our signature 3</p></li></ul></details></div>
</div>
<p><br>
Now use the table from Alexandrov <em>et al.</em> to identify which mutational
processes our three generated signatures have been associated with.</p>
<p><img alt="image" src="../images/mutational_processes.png" /> <strong>Figure 3.</strong> The 21 signatures identified
are indicated as rows with the number corresponding to Figure 2 on the left. The
types of tumours used in the analysis are listed as columns. A green dot at the
intersection of a signature and tumour indicates the signature was identified in
that sample type. Where verified the mutational process is listed on the right.
<em>Source</em>: Alexandrov <em>et al.</em> Nature 2013</p>
<p><br></p>
<div class="admonition note">
<p class="admonition-title">Question</p>
<p>What mutational mechanisms have been associated with the signatures that
you have generated?</p>
<div class="admonition success"><details><summary><strong>Answer</strong></summary><ul><li><p>Our signature 1 (AS7) is associated with Ultraviolet radiation
    damage to DNA. This has previously been identified in Head and Neck
    and Melanoma cancer samples.</p></li><li><p>Our signature 2 (AS 13) is associated with the activity of
    anti-viral APOBEC enzymes. This has previously been seen in Breast
    and Bladder cancer samples.</p></li><li><p>Our signature 3 (AS3) is associated with BRCA1 and BRCA2 mutations,
    i.e. the homologous recombination repair pathway not working
    properly. This has been seen in Breast, Ovarian and Pancreas cancer
    samples.</p></li></ul></details></div>
</div>
<p><br>
Now we can plot out the results for the individual samples in our
dataset to show what proportion of their mutations have been assigned to
each of the signatures.</p>
<table class="codehilitetable"><tr><td class="linenos"><div class="linenodiv"><pre>1
2
3</pre></div></td><td class="code"><div class="codehilite"><pre><span></span>Cairo<span class="p">(</span>file<span class="o">=</span><span class="s">&quot;PlotSampleContribution3Signatures.pdf&quot;</span><span class="p">,</span> type<span class="o">=</span><span class="s">&quot;pdf&quot;</span><span class="p">,</span> units<span class="o">=</span><span class="s">&quot;in&quot;</span><span class="p">,</span> width<span class="o">=</span><span class="m">9</span><span class="p">,</span> height<span class="o">=</span><span class="m">6</span><span class="p">,</span> dpi<span class="o">=</span><span class="m">72</span><span class="p">)</span>
plotSamples<span class="p">(</span>sigs_nmf<span class="p">,</span> normalize<span class="o">=</span><span class="kc">TRUE</span><span class="p">)</span> <span class="o">+</span> scale_y_continuous<span class="p">(</span>breaks<span class="o">=</span><span class="kp">seq</span><span class="p">(</span><span class="m">0</span><span class="p">,</span> <span class="m">1</span><span class="p">,</span> <span class="m">0.2</span><span class="p">),</span> expand <span class="o">=</span> <span class="kt">c</span><span class="p">(</span><span class="m">0</span><span class="p">,</span><span class="m">0</span><span class="p">))</span><span class="o">+</span> theme<span class="p">(</span>axis.text.x <span class="o">=</span> element_text<span class="p">(</span>size<span class="o">=</span><span class="m">6</span><span class="p">))</span>
dev.off<span class="p">()</span>
</pre></div>
</td></tr></table>

<p>If you don’t have time to carry out the advanced questions you can exit
R and return to the normal terminal command line.</p>
<table class="codehilitetable"><tr><td class="linenos"><div class="linenodiv"><pre>1
2</pre></div></td><td class="code"><div class="codehilite"><pre><span></span><span class="kp">quit</span><span class="p">()</span>
n
</pre></div>
</td></tr></table>

<p><br>
Open the resulting <code class="codehilite">PlotSampleContribution3Signatures.pdf</code>.  </p>
<p>This shows the results for the mutation grouping for each sample. The
samples are listed on the x-axis and the proportion of all mutations for
that sample is shown on the y-axis. The colours of the bars indicate
what proportion of the mutations for that sample were grouped into each
of the signatures. The colour that makes up most of the bar for each
sample is called its &ldquo;major signature&rdquo;.</p>
<p>The data you have been using contains samples from High Grade Serous
Ovarian Carcinomas and Cutaneous Melanoma.
<br></p>
<div class="admonition note">
<p class="admonition-title">Question</p>
<p>Using the major signature found for each sample can you guess which are
ovarian and which are melanoma samples?</p>
<div class="admonition success"><details><summary><strong>Answer</strong></summary><ul><li><p>Samples 9-12 have the majority signature of our signature 1. This is
    the UV signature and so these are likely to be Melanoma samples.</p></li><li><p>Samples 4-8 have the majority signature of our signature 3. This is
    the BRCA signature and these are most likely to be ovarian samples.</p></li><li><p>Samples 1-3 have the majority signature of our signature 2. This is
    the APOBEC signature indicating activity of the anti-viral APOBEC
    enzymes. These are less likely to be from cutaneous melanoma because
    they have very few UV associated mutations although it could
    possibly be from a different subtype. However it is much more likely
    that these will be ovarian tumours as this APOBEC signature has been
    seen in breast tumours which can be similar to ovarian cancers in
    terms of the mutated genes.</p></li></ul></details></div>
</div>
<div class="admonition note">
<p class="admonition-title">Question</p>
<p>This is an open question for discussion at the end of the practical.</p>
<p>How can this analysis be useful for cancer genomics studies?</p>
</div>
<div class="admonition note">
<p class="admonition-title"><em>Advanced exercise</em></p>
<p>Now rerun the process this time using 4 signatures as the solution.</p>
<div class="admonition hint"><details><summary><strong>Hint</strong></summary><p>You don’t have to start back at the beginning but you can jump to
the step where you run the NMF but this time for 4 instead of 3
signatures. Then continue through making the plots. You will need to
change the name of each plot you remake with 4 signatures because
Cairo won’t let you overwrite and existing file.</p></details></div>
<p>Can you find a good match in the set of known signatures for all 4 patterns?</p>
<p>Can you find a verified process for all of the profiles you are seeing?</p>
</div>
<hr />
<h2 id="references">References<a class="headerlink" href="#references" title="Permanent link">&para;</a></h2>
<p>Alexandrov <em>et al.</em> Nature 2013: <br />
<a href="http://www.nature.com/nature/journal/v500/n7463/pdf/nature12477.pdf">http://www.nature.com/nature/journal/v500/n7463/pdf/nature12477.pdf</a></p>
<p>Gehring <em>et al.</em> Bioinformatics 2015: <br />
<a href="http://bioinformatics.oxfordjournals.org/content/early/2015/07/31/bioinformatics.btv408.full">http://bioinformatics.oxfordjournals.org/content/early/2015/07/31/bioinformatics.btv408.full</a></p>
                
                  
                
              
              
                
              
            </article>
          </div>
        </div>
      </main>
      
        
<footer class="md-footer">
  
    <div class="md-footer-nav">
      <nav class="md-footer-nav__inner md-grid">
        
          <a href="../../cancer-module-viz/visu/" title="Variant Visualisation" class="md-flex md-footer-nav__link md-footer-nav__link--prev" rel="prev">
            <div class="md-flex__cell md-flex__cell--shrink">
              <i class="md-icon md-icon--arrow-back md-footer-nav__button"></i>
            </div>
            <div class="md-flex__cell md-flex__cell--stretch md-footer-nav__title">
              <span class="md-flex__ellipsis">
                <span class="md-footer-nav__direction">
                  Previous
                </span>
                Variant Visualisation
              </span>
            </div>
          </a>
        
        
          <a href="../02_intogen/" title="Cohort analysis for the identification of driver genes" class="md-flex md-footer-nav__link md-footer-nav__link--next" rel="next">
            <div class="md-flex__cell md-flex__cell--stretch md-footer-nav__title">
              <span class="md-flex__ellipsis">
                <span class="md-footer-nav__direction">
                  Next
                </span>
                Cohort analysis for the identification of driver genes
              </span>
            </div>
            <div class="md-flex__cell md-flex__cell--shrink">
              <i class="md-icon md-icon--arrow-forward md-footer-nav__button"></i>
            </div>
          </a>
        
      </nav>
    </div>
  
  <div class="md-footer-meta md-typeset">
    <div class="md-footer-meta__inner md-grid">
      <div class="md-footer-copyright">
        
          <div class="md-footer-copyright__highlight">
            &copy; The authors; 2017-
          </div>
        
        powered by
        <a href="http://www.mkdocs.org" title="MkDocs">MkDocs</a>
        and
        <a href="http://squidfunk.github.io/mkdocs-material/" title="Material for MkDocs">
          Material for MkDocs</a>
      </div>
      
        
      
    </div>
  </div>
</footer>
      
    </div>
    
      <script src="../../../assets/javascripts/application-b30566c560.js"></script>
      
      
      <script>app.initialize({url:{base:"../../.."}})</script>
      
        <script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.1/MathJax.js?config=TeX-AMS-MML_HTMLorMML"></script>
      
        <script src="../../../js/details.js"></script>
      
    
    
      
    
  </body>
</html>